{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "XUZS4GUkHIEG",
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "# Подготовка файлов и программ\n",
    "\n",
    "Некоторые программы ставятся дважды из разных источников для демонстрации различных вариантов в случае неполадок. Для оптимизации времени работы необходимо удалить лишний код."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "id": "lrtRbP3fAvj8",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "pip install -q condacolab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "id": "jQ1Oqzb9BXoG",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "import condacolab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "q5FUOHCgBZWp",
    "outputId": "9ceec8e3-eb34-4048-f424-29631774cb6c",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "✨🍰✨ Everything looks OK!\n"
     ]
    }
   ],
   "source": [
    "condacolab.install()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "XnbIt77YBOT9",
    "outputId": "39ce240b-5f71-4dcf-c353-febc2b94d364",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
      "Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /usr/local\n",
      "\n",
      "  added / updated specs:\n",
      "    - seqtk\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    ca-certificates-2021.10.8  |       ha878542_0         139 KB  conda-forge\n",
      "    certifi-2021.10.8          |   py37h89c1867_2         145 KB  conda-forge\n",
      "    conda-4.12.0               |   py37h89c1867_0         1.0 MB  conda-forge\n",
      "    seqtk-1.3                  |       h5bf99c6_3          38 KB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         1.3 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  seqtk              bioconda/linux-64::seqtk-1.3-h5bf99c6_3\n",
      "\n",
      "The following packages will be UPDATED:\n",
      "\n",
      "  ca-certificates                      2020.12.5-ha878542_0 --> 2021.10.8-ha878542_0\n",
      "  certifi                          2020.12.5-py37h89c1867_1 --> 2021.10.8-py37h89c1867_2\n",
      "  conda                                4.9.2-py37h89c1867_0 --> 4.12.0-py37h89c1867_0\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "certifi-2021.10.8    | 145 KB    | : 100% 1.0/1 [00:00<00:00,  6.56it/s]                \n",
      "seqtk-1.3            | 38 KB     | : 100% 1.0/1 [00:00<00:00, 14.78it/s]\n",
      "ca-certificates-2021 | 139 KB    | : 100% 1.0/1 [00:00<00:00, 12.76it/s]\n",
      "conda-4.12.0         | 1.0 MB    | : 100% 1.0/1 [00:00<00:00,  2.31it/s]\n",
      "Preparing transaction: \\ \b\bdone\n",
      "Verifying transaction: / \b\bdone\n",
      "Executing transaction: \\ \b\bdone\n"
     ]
    }
   ],
   "source": [
    "!conda install -c bioconda seqtk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "sCBV_31qHOik",
    "outputId": "d5196153-b6a1-49cc-d019-c55c637f1a22",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
      "Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /usr/local\n",
      "\n",
      "  added / updated specs:\n",
      "    - trimmomatic\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    openjdk-8.0.312            |       h7f98852_0        97.6 MB  conda-forge\n",
      "    python_abi-3.7             |          2_cp37m           4 KB  conda-forge\n",
      "    trimmomatic-0.39           |       hdfd78af_2         144 KB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:        97.8 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  openjdk            conda-forge/linux-64::openjdk-8.0.312-h7f98852_0\n",
      "  trimmomatic        bioconda/noarch::trimmomatic-0.39-hdfd78af_2\n",
      "\n",
      "The following packages will be UPDATED:\n",
      "\n",
      "  python_abi                                    3.7-1_cp37m --> 3.7-2_cp37m\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "python_abi-3.7       | 4 KB      | : 100% 1.0/1 [00:00<00:00,  6.92it/s]\n",
      "openjdk-8.0.312      | 97.6 MB   | : 100% 1.0/1 [00:15<00:00, 15.53s/it]               \n",
      "trimmomatic-0.39     | 144 KB    | : 100% 1.0/1 [00:00<00:00, 13.88it/s]\n",
      "Preparing transaction: - \b\bdone\n",
      "Verifying transaction: | \b\bdone\n",
      "Executing transaction: - \b\bdone\n"
     ]
    }
   ],
   "source": [
    "!conda install -c bioconda trimmomatic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 1000
    },
    "id": "F0BLWsH2n3Ov",
    "outputId": "f1fafcd6-75a0-4155-e2c3-27598d724b98",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "Collecting macs2\n",
      "  Downloading MACS2-2.2.7.1.tar.gz (2.1 MB)\n",
      "\u001B[K     |████████████████████████████████| 2.1 MB 5.2 MB/s \n",
      "\u001B[?25h  Installing build dependencies ... \u001B[?25l\u001B[?25hdone\n",
      "  Getting requirements to build wheel ... \u001B[?25l\u001B[?25hdone\n",
      "    Preparing wheel metadata ... \u001B[?25l\u001B[?25hdone\n",
      "Collecting intervene\n",
      "  Downloading intervene-0.6.5.tar.gz (5.5 MB)\n",
      "\u001B[K     |████████████████████████████████| 5.5 MB 30.9 MB/s \n",
      "\u001B[?25hCollecting pybedtools\n",
      "  Downloading pybedtools-0.9.0.tar.gz (12.5 MB)\n",
      "\u001B[K     |████████████████████████████████| 12.5 MB 22.2 MB/s \n",
      "\u001B[?25hCollecting matplotlib\n",
      "  Downloading matplotlib-3.5.2-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (11.2 MB)\n",
      "\u001B[K     |████████████████████████████████| 11.2 MB 41.9 MB/s \n",
      "\u001B[?25hCollecting pandas\n",
      "  Downloading pandas-1.3.5-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.3 MB)\n",
      "\u001B[K     |████████████████████████████████| 11.3 MB 32.9 MB/s \n",
      "\u001B[?25hCollecting numpy\n",
      "  Using cached numpy-1.21.6-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (15.7 MB)\n",
      "Collecting scipy\n",
      "  Downloading scipy-1.7.3-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (38.1 MB)\n",
      "\u001B[K     |████████████████████████████████| 38.1 MB 1.4 MB/s \n",
      "\u001B[?25hCollecting seaborn\n",
      "  Downloading seaborn-0.11.2-py3-none-any.whl (292 kB)\n",
      "\u001B[K     |████████████████████████████████| 292 kB 33.9 MB/s \n",
      "\u001B[?25hCollecting kiwisolver>=1.0.1\n",
      "  Downloading kiwisolver-1.4.2-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (1.1 MB)\n",
      "\u001B[K     |████████████████████████████████| 1.1 MB 41.7 MB/s \n",
      "\u001B[?25hCollecting packaging>=20.0\n",
      "  Downloading packaging-21.3-py3-none-any.whl (40 kB)\n",
      "\u001B[K     |████████████████████████████████| 40 kB 4.0 MB/s \n",
      "\u001B[?25hCollecting cycler>=0.10\n",
      "  Downloading cycler-0.11.0-py3-none-any.whl (6.4 kB)\n",
      "Collecting python-dateutil>=2.7\n",
      "  Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)\n",
      "\u001B[K     |████████████████████████████████| 247 kB 3.9 MB/s \n",
      "\u001B[?25hCollecting pyparsing>=2.2.1\n",
      "  Downloading pyparsing-3.0.8-py3-none-any.whl (98 kB)\n",
      "\u001B[K     |████████████████████████████████| 98 kB 7.0 MB/s \n",
      "\u001B[?25hCollecting pillow>=6.2.0\n",
      "  Downloading Pillow-9.1.0-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.3 MB)\n",
      "\u001B[K     |████████████████████████████████| 4.3 MB 28.0 MB/s \n",
      "\u001B[?25hCollecting fonttools>=4.22.0\n",
      "  Downloading fonttools-4.33.3-py3-none-any.whl (930 kB)\n",
      "\u001B[K     |████████████████████████████████| 930 kB 38.2 MB/s \n",
      "\u001B[?25hCollecting typing-extensions\n",
      "  Downloading typing_extensions-4.2.0-py3-none-any.whl (24 kB)\n",
      "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.7/site-packages (from python-dateutil>=2.7->matplotlib->intervene) (1.15.0)\n",
      "Collecting pytz>=2017.3\n",
      "  Downloading pytz-2022.1-py2.py3-none-any.whl (503 kB)\n",
      "\u001B[K     |████████████████████████████████| 503 kB 36.7 MB/s \n",
      "\u001B[?25hCollecting pysam\n",
      "  Downloading pysam-0.19.0-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (15.0 MB)\n",
      "\u001B[K     |████████████████████████████████| 15.0 MB 22.5 MB/s \n",
      "\u001B[?25hBuilding wheels for collected packages: intervene, macs2, pybedtools\n",
      "  Building wheel for intervene (setup.py) ... \u001B[?25l\u001B[?25hdone\n",
      "  Created wheel for intervene: filename=intervene-0.6.5-py3-none-any.whl size=5546118 sha256=c700063c112ec98ffefd42af248c2dcbdaac93dc25f463a5b6f5f461cb892f54\n",
      "  Stored in directory: /root/.cache/pip/wheels/f3/d2/c1/18fa6122c36fdc731af4f16c59c356606b12a890a578294044\n",
      "  Building wheel for macs2 (PEP 517) ... \u001B[?25l\u001B[?25hdone\n",
      "  Created wheel for macs2: filename=MACS2-2.2.7.1-cp37-cp37m-linux_x86_64.whl size=1790949 sha256=6c9263798729db08909de3c454162334598b8cafba3106e456517626ed2a7c38\n",
      "  Stored in directory: /root/.cache/pip/wheels/4e/34/10/f6222945f7c5b091e34b058a01a842ba9b1d960f2ab0c8e16f\n",
      "  Building wheel for pybedtools (setup.py) ... \u001B[?25l\u001B[?25hdone\n",
      "  Created wheel for pybedtools: filename=pybedtools-0.9.0-cp37-cp37m-linux_x86_64.whl size=13654299 sha256=15a291f11eaefc70411bdbbdb69b0afaf89660f9dc353eac8d518cc4a9e1d646\n",
      "  Stored in directory: /root/.cache/pip/wheels/7a/44/0d/3a7449885adaf8ebb157da8c3c834a712f48b3b3b84ba51dda\n",
      "Successfully built intervene macs2 pybedtools\n",
      "Installing collected packages: typing-extensions, pyparsing, pytz, python-dateutil, pillow, packaging, numpy, kiwisolver, fonttools, cycler, scipy, pysam, pandas, matplotlib, seaborn, pybedtools, macs2, intervene\n",
      "Successfully installed cycler-0.11.0 fonttools-4.33.3 intervene-0.6.5 kiwisolver-1.4.2 macs2-2.2.7.1 matplotlib-3.5.2 numpy-1.21.6 packaging-21.3 pandas-1.3.5 pillow-9.1.0 pybedtools-0.9.0 pyparsing-3.0.8 pysam-0.19.0 python-dateutil-2.8.2 pytz-2022.1 scipy-1.7.3 seaborn-0.11.2 typing-extensions-4.2.0\n"
     ]
    },
    {
     "output_type": "display_data",
     "data": {
      "application/vnd.colab-display-data+json": {
       "pip_warning": {
        "packages": [
         "cycler",
         "dateutil",
         "kiwisolver",
         "numpy"
        ]
       }
      }
     },
     "metadata": {}
    }
   ],
   "source": [
    "!pip install macs2 intervene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "Ce6hiSTdkarU",
    "outputId": "8c75c333-ad00-425c-c0bd-f34ecf7570e7",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "Reading package lists... Done\n",
      "Building dependency tree       \n",
      "Reading state information... Done\n",
      "The following packages were automatically installed and are no longer required:\n",
      "  libnvidia-common-460 nsight-compute-2020.2.0\n",
      "Use 'apt autoremove' to remove them.\n",
      "The following NEW packages will be installed:\n",
      "  bedtools\n",
      "0 upgraded, 1 newly installed, 0 to remove and 42 not upgraded.\n",
      "Need to get 577 kB of archives.\n",
      "After this operation, 2,040 kB of additional disk space will be used.\n",
      "Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bedtools amd64 2.26.0+dfsg-5 [577 kB]\n",
      "Fetched 577 kB in 1s (721 kB/s)\n",
      "Selecting previously unselected package bedtools.\n",
      "(Reading database ... 155203 files and directories currently installed.)\n",
      "Preparing to unpack .../bedtools_2.26.0+dfsg-5_amd64.deb ...\n",
      "Unpacking bedtools (2.26.0+dfsg-5) ...\n",
      "Setting up bedtools (2.26.0+dfsg-5) ...\n"
     ]
    }
   ],
   "source": [
    "!apt-get install bedtools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "AUML3dvsqT9E",
    "outputId": "f9f998f8-6c5b-4771-ad4f-90960410a4a3",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "Reading package lists... Done\n",
      "Building dependency tree       \n",
      "Reading state information... Done\n",
      "The following packages were automatically installed and are no longer required:\n",
      "  libnvidia-common-460 nsight-compute-2020.2.0\n",
      "Use 'apt autoremove' to remove them.\n",
      "The following additional packages will be installed:\n",
      "  cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc\n",
      "  python-asn1crypto python-avro python-cachecontrol python-certifi\n",
      "  python-cffi-backend python-chardet python-cryptography python-enum34\n",
      "  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n",
      "  python-mistune python-openssl python-pkg-resources python-pyparsing\n",
      "  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n",
      "  python-schema-salad python-shellescape python-six python-sparqlwrapper\n",
      "  python-typing python-urllib3 python-webencodings\n",
      "Suggested packages:\n",
      "  bowtie2-examples python-cryptography-doc python-cryptography-vectors\n",
      "  python-enum34-doc python-genshi python-lxml python-lockfile-doc\n",
      "  python-openssl-doc python-openssl-dbg python-setuptools python-pyparsing-doc\n",
      "  python-rdflib-doc python-rdflib-tools python-socks python-ntlm\n",
      "The following NEW packages will be installed:\n",
      "  bowtie2 cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc\n",
      "  python-asn1crypto python-avro python-cachecontrol python-certifi\n",
      "  python-cffi-backend python-chardet python-cryptography python-enum34\n",
      "  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n",
      "  python-mistune python-openssl python-pkg-resources python-pyparsing\n",
      "  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n",
      "  python-schema-salad python-shellescape python-six python-sparqlwrapper\n",
      "  python-typing python-urllib3 python-webencodings samtools\n",
      "0 upgraded, 36 newly installed, 0 to remove and 42 not upgraded.\n",
      "Need to get 9,743 kB of archives.\n",
      "After this operation, 46.1 MB of additional disk space will be used.\n",
      "Get:1 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pkg-resources all 39.0.1-2 [128 kB]\n",
      "Get:2 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-six all 1.11.0-2 [11.3 kB]\n",
      "Get:3 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-isodate all 0.6.0-1 [23.8 kB]\n",
      "Get:4 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pyparsing all 2.2.0+dfsg1-2 [52.1 kB]\n",
      "Get:5 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib all 4.2.1-2 [237 kB]\n",
      "Get:6 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-certifi all 2018.1.18-2 [144 kB]\n",
      "Get:7 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-chardet all 3.0.4-1 [80.3 kB]\n",
      "Get:8 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-idna all 2.6-1 [32.4 kB]\n",
      "Get:9 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-urllib3 all 1.22-1ubuntu0.18.04.2 [86.0 kB]\n",
      "Get:10 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-requests all 2.18.4-2ubuntu0.1 [58.5 kB]\n",
      "Get:11 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-ruamel.yaml amd64 0.15.34-1 [183 kB]\n",
      "Get:12 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-cachecontrol all 0.11.7-1 [15.2 kB]\n",
      "Get:13 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-lockfile all 1:0.12.2-2 [14.6 kB]\n",
      "Get:14 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-mistune all 0.8.3-2 [13.8 kB]\n",
      "Get:15 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib-jsonld all 0.4.0-3 [12.2 kB]\n",
      "Get:16 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-typing all 3.6.2-1 [22.2 kB]\n",
      "Get:17 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-avro all 1.8.2+dfsg-1 [23.2 kB]\n",
      "Get:18 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libjs-bootstrap all 3.3.7+dfsg-2 [228 kB]\n",
      "Get:19 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-schema-salad all 2.6.20171201034858-3 [260 kB]\n",
      "Get:20 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-shellescape all 3.4.1-2 [4,432 B]\n",
      "Get:21 http://archive.ubuntu.com/ubuntu bionic/universe amd64 cwltool all 1.0.20180302231433-1 [143 kB]\n",
      "Get:22 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libhts2 amd64 1.7-2 [300 kB]\n",
      "Get:23 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs-doc all 8.10.0~dfsg-2ubuntu0.4 [752 kB]\n",
      "Get:24 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-asn1crypto all 0.24.0-1 [72.7 kB]\n",
      "Get:25 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-cffi-backend amd64 1.11.5-1 [63.4 kB]\n",
      "Get:26 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-enum34 all 1.1.6-2 [34.8 kB]\n",
      "Get:27 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-ipaddress all 1.0.17-1 [18.2 kB]\n",
      "Get:28 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-cryptography amd64 2.1.4-1ubuntu1.4 [276 kB]\n",
      "Get:29 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-webencodings all 0.5-2 [10.3 kB]\n",
      "Get:30 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-html5lib all 0.999999999-1 [83.6 kB]\n",
      "Get:31 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-openssl all 17.5.0-1ubuntu1 [41.3 kB]\n",
      "Get:32 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-sparqlwrapper all 1.7.6-2 [21.8 kB]\n",
      "Get:33 http://archive.ubuntu.com/ubuntu bionic/universe amd64 samtools amd64 1.7-1 [288 kB]\n",
      "Get:34 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bowtie2 amd64 2.3.4.1-1 [1,174 kB]\n",
      "Get:35 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libc-ares2 amd64 1.14.0-1ubuntu0.1 [37.5 kB]\n",
      "Get:36 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs amd64 8.10.0~dfsg-2ubuntu0.4 [4,796 kB]\n",
      "Fetched 9,743 kB in 2s (6,207 kB/s)\n",
      "Extracting templates from packages: 100%\n",
      "Selecting previously unselected package python-pkg-resources.\n",
      "(Reading database ... 155254 files and directories currently installed.)\n",
      "Preparing to unpack .../00-python-pkg-resources_39.0.1-2_all.deb ...\n",
      "Unpacking python-pkg-resources (39.0.1-2) ...\n",
      "Selecting previously unselected package python-six.\n",
      "Preparing to unpack .../01-python-six_1.11.0-2_all.deb ...\n",
      "Unpacking python-six (1.11.0-2) ...\n",
      "Selecting previously unselected package python-isodate.\n",
      "Preparing to unpack .../02-python-isodate_0.6.0-1_all.deb ...\n",
      "Unpacking python-isodate (0.6.0-1) ...\n",
      "Selecting previously unselected package python-pyparsing.\n",
      "Preparing to unpack .../03-python-pyparsing_2.2.0+dfsg1-2_all.deb ...\n",
      "Unpacking python-pyparsing (2.2.0+dfsg1-2) ...\n",
      "Selecting previously unselected package python-rdflib.\n",
      "Preparing to unpack .../04-python-rdflib_4.2.1-2_all.deb ...\n",
      "Unpacking python-rdflib (4.2.1-2) ...\n",
      "Selecting previously unselected package python-certifi.\n",
      "Preparing to unpack .../05-python-certifi_2018.1.18-2_all.deb ...\n",
      "Unpacking python-certifi (2018.1.18-2) ...\n",
      "Selecting previously unselected package python-chardet.\n",
      "Preparing to unpack .../06-python-chardet_3.0.4-1_all.deb ...\n",
      "Unpacking python-chardet (3.0.4-1) ...\n",
      "Selecting previously unselected package python-idna.\n",
      "Preparing to unpack .../07-python-idna_2.6-1_all.deb ...\n",
      "Unpacking python-idna (2.6-1) ...\n",
      "Selecting previously unselected package python-urllib3.\n",
      "Preparing to unpack .../08-python-urllib3_1.22-1ubuntu0.18.04.2_all.deb ...\n",
      "Unpacking python-urllib3 (1.22-1ubuntu0.18.04.2) ...\n",
      "Selecting previously unselected package python-requests.\n",
      "Preparing to unpack .../09-python-requests_2.18.4-2ubuntu0.1_all.deb ...\n",
      "Unpacking python-requests (2.18.4-2ubuntu0.1) ...\n",
      "Selecting previously unselected package python-ruamel.yaml.\n",
      "Preparing to unpack .../10-python-ruamel.yaml_0.15.34-1_amd64.deb ...\n",
      "Unpacking python-ruamel.yaml (0.15.34-1) ...\n",
      "Selecting previously unselected package python-cachecontrol.\n",
      "Preparing to unpack .../11-python-cachecontrol_0.11.7-1_all.deb ...\n",
      "Unpacking python-cachecontrol (0.11.7-1) ...\n",
      "Selecting previously unselected package python-lockfile.\n",
      "Preparing to unpack .../12-python-lockfile_1%3a0.12.2-2_all.deb ...\n",
      "Unpacking python-lockfile (1:0.12.2-2) ...\n",
      "Selecting previously unselected package python-mistune.\n",
      "Preparing to unpack .../13-python-mistune_0.8.3-2_all.deb ...\n",
      "Unpacking python-mistune (0.8.3-2) ...\n",
      "Selecting previously unselected package python-rdflib-jsonld.\n",
      "Preparing to unpack .../14-python-rdflib-jsonld_0.4.0-3_all.deb ...\n",
      "Unpacking python-rdflib-jsonld (0.4.0-3) ...\n",
      "Selecting previously unselected package python-typing.\n",
      "Preparing to unpack .../15-python-typing_3.6.2-1_all.deb ...\n",
      "Unpacking python-typing (3.6.2-1) ...\n",
      "Selecting previously unselected package python-avro.\n",
      "Preparing to unpack .../16-python-avro_1.8.2+dfsg-1_all.deb ...\n",
      "Unpacking python-avro (1.8.2+dfsg-1) ...\n",
      "Selecting previously unselected package libjs-bootstrap.\n",
      "Preparing to unpack .../17-libjs-bootstrap_3.3.7+dfsg-2_all.deb ...\n",
      "Unpacking libjs-bootstrap (3.3.7+dfsg-2) ...\n",
      "Selecting previously unselected package python-schema-salad.\n",
      "Preparing to unpack .../18-python-schema-salad_2.6.20171201034858-3_all.deb ...\n",
      "Unpacking python-schema-salad (2.6.20171201034858-3) ...\n",
      "Selecting previously unselected package python-shellescape.\n",
      "Preparing to unpack .../19-python-shellescape_3.4.1-2_all.deb ...\n",
      "Unpacking python-shellescape (3.4.1-2) ...\n",
      "Selecting previously unselected package cwltool.\n",
      "Preparing to unpack .../20-cwltool_1.0.20180302231433-1_all.deb ...\n",
      "Unpacking cwltool (1.0.20180302231433-1) ...\n",
      "Selecting previously unselected package libhts2:amd64.\n",
      "Preparing to unpack .../21-libhts2_1.7-2_amd64.deb ...\n",
      "Unpacking libhts2:amd64 (1.7-2) ...\n",
      "Selecting previously unselected package nodejs-doc.\n",
      "Preparing to unpack .../22-nodejs-doc_8.10.0~dfsg-2ubuntu0.4_all.deb ...\n",
      "Unpacking nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
      "Selecting previously unselected package python-asn1crypto.\n",
      "Preparing to unpack .../23-python-asn1crypto_0.24.0-1_all.deb ...\n",
      "Unpacking python-asn1crypto (0.24.0-1) ...\n",
      "Selecting previously unselected package python-cffi-backend.\n",
      "Preparing to unpack .../24-python-cffi-backend_1.11.5-1_amd64.deb ...\n",
      "Unpacking python-cffi-backend (1.11.5-1) ...\n",
      "Selecting previously unselected package python-enum34.\n",
      "Preparing to unpack .../25-python-enum34_1.1.6-2_all.deb ...\n",
      "Unpacking python-enum34 (1.1.6-2) ...\n",
      "Selecting previously unselected package python-ipaddress.\n",
      "Preparing to unpack .../26-python-ipaddress_1.0.17-1_all.deb ...\n",
      "Unpacking python-ipaddress (1.0.17-1) ...\n",
      "Selecting previously unselected package python-cryptography.\n",
      "Preparing to unpack .../27-python-cryptography_2.1.4-1ubuntu1.4_amd64.deb ...\n",
      "Unpacking python-cryptography (2.1.4-1ubuntu1.4) ...\n",
      "Selecting previously unselected package python-webencodings.\n",
      "Preparing to unpack .../28-python-webencodings_0.5-2_all.deb ...\n",
      "Unpacking python-webencodings (0.5-2) ...\n",
      "Selecting previously unselected package python-html5lib.\n",
      "Preparing to unpack .../29-python-html5lib_0.999999999-1_all.deb ...\n",
      "Unpacking python-html5lib (0.999999999-1) ...\n",
      "Selecting previously unselected package python-openssl.\n",
      "Preparing to unpack .../30-python-openssl_17.5.0-1ubuntu1_all.deb ...\n",
      "Unpacking python-openssl (17.5.0-1ubuntu1) ...\n",
      "Selecting previously unselected package python-sparqlwrapper.\n",
      "Preparing to unpack .../31-python-sparqlwrapper_1.7.6-2_all.deb ...\n",
      "Unpacking python-sparqlwrapper (1.7.6-2) ...\n",
      "Selecting previously unselected package samtools.\n",
      "Preparing to unpack .../32-samtools_1.7-1_amd64.deb ...\n",
      "Unpacking samtools (1.7-1) ...\n",
      "Selecting previously unselected package bowtie2.\n",
      "Preparing to unpack .../33-bowtie2_2.3.4.1-1_amd64.deb ...\n",
      "Unpacking bowtie2 (2.3.4.1-1) ...\n",
      "Selecting previously unselected package libc-ares2:amd64.\n",
      "Preparing to unpack .../34-libc-ares2_1.14.0-1ubuntu0.1_amd64.deb ...\n",
      "Unpacking libc-ares2:amd64 (1.14.0-1ubuntu0.1) ...\n",
      "Selecting previously unselected package nodejs.\n",
      "Preparing to unpack .../35-nodejs_8.10.0~dfsg-2ubuntu0.4_amd64.deb ...\n",
      "Unpacking nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
      "Setting up python-idna (2.6-1) ...\n",
      "Setting up nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
      "Setting up python-asn1crypto (0.24.0-1) ...\n",
      "Setting up python-mistune (0.8.3-2) ...\n",
      "Setting up python-certifi (2018.1.18-2) ...\n",
      "Setting up python-avro (1.8.2+dfsg-1) ...\n",
      "Setting up python-pkg-resources (39.0.1-2) ...\n",
      "Setting up bowtie2 (2.3.4.1-1) ...\n",
      "Setting up python-ruamel.yaml (0.15.34-1) ...\n",
      "Setting up python-cffi-backend (1.11.5-1) ...\n",
      "Setting up python-six (1.11.0-2) ...\n",
      "Setting up python-isodate (0.6.0-1) ...\n",
      "Setting up python-shellescape (3.4.1-2) ...\n",
      "Setting up python-enum34 (1.1.6-2) ...\n",
      "Setting up libjs-bootstrap (3.3.7+dfsg-2) ...\n",
      "Setting up python-lockfile (1:0.12.2-2) ...\n",
      "Setting up python-ipaddress (1.0.17-1) ...\n",
      "Setting up libc-ares2:amd64 (1.14.0-1ubuntu0.1) ...\n",
      "Setting up libhts2:amd64 (1.7-2) ...\n",
      "Setting up python-webencodings (0.5-2) ...\n",
      "Setting up python-typing (3.6.2-1) ...\n",
      "Setting up python-pyparsing (2.2.0+dfsg1-2) ...\n",
      "Setting up python-urllib3 (1.22-1ubuntu0.18.04.2) ...\n",
      "Setting up python-chardet (3.0.4-1) ...\n",
      "Setting up nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
      "update-alternatives: using /usr/bin/nodejs to provide /usr/bin/js (js) in auto mode\n",
      "Setting up python-rdflib (4.2.1-2) ...\n",
      "Setting up python-sparqlwrapper (1.7.6-2) ...\n",
      "Setting up python-rdflib-jsonld (0.4.0-3) ...\n",
      "Setting up python-cryptography (2.1.4-1ubuntu1.4) ...\n",
      "Setting up python-requests (2.18.4-2ubuntu0.1) ...\n",
      "Setting up python-cachecontrol (0.11.7-1) ...\n",
      "Setting up python-schema-salad (2.6.20171201034858-3) ...\n",
      "Setting up python-html5lib (0.999999999-1) ...\n",
      "Setting up samtools (1.7-1) ...\n",
      "Setting up python-openssl (17.5.0-1ubuntu1) ...\n",
      "Setting up cwltool (1.0.20180302231433-1) ...\n",
      "Processing triggers for libc-bin (2.27-3ubuntu1.3) ...\n",
      "/sbin/ldconfig.real: file /usr/local/lib/libnet.so is truncated\n",
      "\n",
      "/sbin/ldconfig.real: file /usr/local/lib/libjli.so is truncated\n",
      "\n",
      "/sbin/ldconfig.real: file /usr/local/lib/libjawt.so is truncated\n",
      "\n",
      "/sbin/ldconfig.real: /usr/local/lib/python3.7/dist-packages/ideep4py/lib/libmkldnn.so.0 is not a symbolic link\n",
      "\n",
      "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n"
     ]
    }
   ],
   "source": [
    "!apt-get install bowtie2 samtools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "JgTq4lFjq93N",
    "outputId": "daafacca-7b73-4923-aaae-7d87fe25d5e1",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
      "Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /usr/local\n",
      "\n",
      "  added / updated specs:\n",
      "    - bowtie2\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    bowtie2-2.3.5.1            |   py37he513fc3_0        12.0 MB  bioconda\n",
      "    perl-5.32.1                | 0_h7f98852_perl5        14.5 MB  conda-forge\n",
      "    tbb-2020.2                 |       h4bd325d_4         1.5 MB  conda-forge\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:        27.9 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  bowtie2            bioconda/linux-64::bowtie2-2.3.5.1-py37he513fc3_0\n",
      "  perl               conda-forge/linux-64::perl-5.32.1-0_h7f98852_perl5\n",
      "  tbb                conda-forge/linux-64::tbb-2020.2-h4bd325d_4\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "perl-5.32.1          | 14.5 MB   | : 100% 1.0/1 [00:04<00:00,  4.14s/it]               \n",
      "bowtie2-2.3.5.1      | 12.0 MB   | : 100% 1.0/1 [00:03<00:00,  3.10s/it]               \n",
      "tbb-2020.2           | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  2.70it/s]\n",
      "Preparing transaction: / \b\b- \b\bdone\n",
      "Verifying transaction: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
      "Executing transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n"
     ]
    }
   ],
   "source": [
    "!conda install -c bioconda bowtie2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "vcr5CjibHTgF",
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "## Установка FastQC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "62l1Q2ymbENA",
    "outputId": "f4302c39-2764-41b4-ec45-139d6868ebf0",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "openjdk version \"1.8.0_312\"\n",
      "OpenJDK Runtime Environment (Zulu 8.58.0.13-CA-linux64) (build 1.8.0_312-b07)\n",
      "OpenJDK 64-Bit Server VM (Zulu 8.58.0.13-CA-linux64) (build 25.312-b07, mixed mode)\n"
     ]
    }
   ],
   "source": [
    "!java -version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "8bpgwFgcbgRd",
    "outputId": "ec3bc6d3-3cac-4cbb-ccde-5ab5601b366b",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "--2022-03-07 17:59:21--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
      "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
      "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 10249221 (9.8M) [application/zip]\n",
      "Saving to: ‘fastqc_v0.11.9.zip’\n",
      "\n",
      "fastqc_v0.11.9.zip  100%[===================>]   9.77M  3.90MB/s    in 2.5s    \n",
      "\n",
      "2022-03-07 17:59:24 (3.90 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "yTcTjOTJblcm",
    "outputId": "0616510a-7850-4fe0-f17e-d90014afe32d",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "Archive:  fastqc_v0.11.9.zip\n",
      "  inflating: FastQC/cisd-jhdf5.jar   \n",
      "   creating: FastQC/Configuration/\n",
      "  inflating: FastQC/Configuration/adapter_list.txt  \n",
      "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
      "  inflating: FastQC/Configuration/limits.txt  \n",
      "  inflating: FastQC/fastqc           \n",
      "  inflating: FastQC/fastqc_icon.ico  \n",
      "   creating: FastQC/Help/\n",
      "   creating: FastQC/Help/1 Introduction/\n",
      "   creating: FastQC/Help/1 Introduction/.svn/\n",
      "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
      "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
      "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
      "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
      "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
      "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
      "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
      "   creating: FastQC/Help/2 Basic Operations/\n",
      "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
      "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
      "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
      "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
      "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
      "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
      "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
      "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
      "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
      "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
      "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
      "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
      "   creating: FastQC/Help/3 Analysis Modules/\n",
      "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
      "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
      "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
      "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
      "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
      "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
      "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
      "  inflating: FastQC/INSTALL.txt      \n",
      "  inflating: FastQC/jbzip2-0.9.jar   \n",
      "  inflating: FastQC/LICENSE          \n",
      "  inflating: FastQC/LICENSE.txt      \n",
      "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
      "   creating: FastQC/net/\n",
      "   creating: FastQC/net/sourceforge/\n",
      "   creating: FastQC/net/sourceforge/iharder/\n",
      "   creating: FastQC/net/sourceforge/iharder/base64/\n",
      "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
      "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
      "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
      "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
      "   creating: FastQC/org/\n",
      "   creating: FastQC/org/apache/\n",
      "   creating: FastQC/org/apache/commons/\n",
      "   creating: FastQC/org/apache/commons/math3/\n",
      "   creating: FastQC/org/apache/commons/math3/analysis/\n",
      "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
      "   creating: FastQC/org/apache/commons/math3/distribution/\n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
      "   creating: FastQC/org/apache/commons/math3/exception/\n",
      "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
      "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
      "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
      "   creating: FastQC/org/apache/commons/math3/random/\n",
      "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
      "   creating: FastQC/org/apache/commons/math3/special/\n",
      "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
      "   creating: FastQC/org/apache/commons/math3/util/\n",
      "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
      "  inflating: FastQC/README.md        \n",
      "  inflating: FastQC/README.txt       \n",
      "  inflating: FastQC/RELEASE_NOTES.txt  \n",
      "  inflating: FastQC/run_fastqc.bat   \n",
      "  inflating: FastQC/sam-1.103.jar    \n",
      "   creating: FastQC/Templates/\n",
      "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
      "  inflating: FastQC/Templates/header_template.html  \n",
      "   creating: FastQC/Templates/Icons/\n",
      " extracting: FastQC/Templates/Icons/error.png  \n",
      " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
      " extracting: FastQC/Templates/Icons/tick.png  \n",
      " extracting: FastQC/Templates/Icons/warning.png  \n",
      "   creating: FastQC/uk/\n",
      "   creating: FastQC/uk/ac/\n",
      "   creating: FastQC/uk/ac/babraham/\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
      " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
      " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
      " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
      " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
      " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n"
     ]
    }
   ],
   "source": [
    "!unzip fastqc_v0.11.9.zip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "id": "S79LH0aXbvLx",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "!chmod a+x FastQC/fastqc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "PsLuZDt4b7jT",
    "outputId": "bfecb352-d387-4b56-874d-e892665a595d",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "\n",
      "            FastQC - A high throughput sequence QC analysis tool\n",
      "\n",
      "SYNOPSIS\n",
      "\n",
      "\tfastqc seqfile1 seqfile2 .. seqfileN\n",
      "\n",
      "    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] \n",
      "           [-c contaminant file] seqfile1 .. seqfileN\n",
      "\n",
      "DESCRIPTION\n",
      "\n",
      "    FastQC reads a set of sequence files and produces from each one a quality\n",
      "    control report consisting of a number of different modules, each one of \n",
      "    which will help to identify a different potential type of problem in your\n",
      "    data.\n",
      "    \n",
      "    If no files to process are specified on the command line then the program\n",
      "    will start as an interactive graphical application.  If files are provided\n",
      "    on the command line then the program will run with no user interaction\n",
      "    required.  In this mode it is suitable for inclusion into a standardised\n",
      "    analysis pipeline.\n",
      "    \n",
      "    The options for the program as as follows:\n",
      "    \n",
      "    -h --help       Print this help file and exit\n",
      "    \n",
      "    -v --version    Print the version of the program and exit\n",
      "    \n",
      "    -o --outdir     Create all output files in the specified output directory.\n",
      "                    Please note that this directory must exist as the program\n",
      "                    will not create it.  If this option is not set then the \n",
      "                    output file for each sequence file is created in the same\n",
      "                    directory as the sequence file which was processed.\n",
      "                    \n",
      "    --casava        Files come from raw casava output. Files in the same sample\n",
      "                    group (differing only by the group number) will be analysed\n",
      "                    as a set rather than individually. Sequences with the filter\n",
      "                    flag set in the header will be excluded from the analysis.\n",
      "                    Files must have the same names given to them by casava\n",
      "                    (including being gzipped and ending with .gz) otherwise they\n",
      "                    won't be grouped together correctly.\n",
      "                    \n",
      "    --nano          Files come from nanopore sequences and are in fast5 format. In\n",
      "                    this mode you can pass in directories to process and the program\n",
      "                    will take in all fast5 files within those directories and produce\n",
      "                    a single output file from the sequences found in all files.                    \n",
      "                    \n",
      "    --nofilter      If running with --casava then don't remove read flagged by\n",
      "                    casava as poor quality when performing the QC analysis.\n",
      "                   \n",
      "    --extract       If set then the zipped output file will be uncompressed in\n",
      "                    the same directory after it has been created.  By default\n",
      "                    this option will be set if fastqc is run in non-interactive\n",
      "                    mode.\n",
      "                    \n",
      "    -j --java       Provides the full path to the java binary you want to use to\n",
      "                    launch fastqc. If not supplied then java is assumed to be in\n",
      "                    your path.\n",
      "                   \n",
      "    --noextract     Do not uncompress the output file after creating it.  You\n",
      "                    should set this option if you do not wish to uncompress\n",
      "                    the output when running in non-interactive mode.\n",
      "                    \n",
      "    --nogroup       Disable grouping of bases for reads >50bp. All reports will\n",
      "                    show data for every base in the read.  WARNING: Using this\n",
      "                    option will cause fastqc to crash and burn if you use it on\n",
      "                    really long reads, and your plots may end up a ridiculous size.\n",
      "                    You have been warned!\n",
      "                    \n",
      "    --min_length    Sets an artificial lower limit on the length of the sequence\n",
      "                    to be shown in the report.  As long as you set this to a value\n",
      "                    greater or equal to your longest read length then this will be\n",
      "                    the sequence length used to create your read groups.  This can\n",
      "                    be useful for making directly comaparable statistics from \n",
      "                    datasets with somewhat variable read lengths.\n",
      "                    \n",
      "    -f --format     Bypasses the normal sequence file format detection and\n",
      "                    forces the program to use the specified format.  Valid\n",
      "                    formats are bam,sam,bam_mapped,sam_mapped and fastq\n",
      "                    \n",
      "    -t --threads    Specifies the number of files which can be processed\n",
      "                    simultaneously.  Each thread will be allocated 250MB of\n",
      "                    memory so you shouldn't run more threads than your\n",
      "                    available memory will cope with, and not more than\n",
      "                    6 threads on a 32 bit machine\n",
      "                  \n",
      "    -c              Specifies a non-default file which contains the list of\n",
      "    --contaminants  contaminants to screen overrepresented sequences against.\n",
      "                    The file must contain sets of named contaminants in the\n",
      "                    form name[tab]sequence.  Lines prefixed with a hash will\n",
      "                    be ignored.\n",
      "\n",
      "    -a              Specifies a non-default file which contains the list of\n",
      "    --adapters      adapter sequences which will be explicity searched against\n",
      "                    the library. The file must contain sets of named adapters\n",
      "                    in the form name[tab]sequence.  Lines prefixed with a hash\n",
      "                    will be ignored.\n",
      "                    \n",
      "    -l              Specifies a non-default file which contains a set of criteria\n",
      "    --limits        which will be used to determine the warn/error limits for the\n",
      "                    various modules.  This file can also be used to selectively \n",
      "                    remove some modules from the output all together.  The format\n",
      "                    needs to mirror the default limits.txt file found in the\n",
      "                    Configuration folder.\n",
      "                    \n",
      "   -k --kmers       Specifies the length of Kmer to look for in the Kmer content\n",
      "                    module. Specified Kmer length must be between 2 and 10. Default\n",
      "                    length is 7 if not specified.\n",
      "                    \n",
      "   -q --quiet       Supress all progress messages on stdout and only report errors.\n",
      "   \n",
      "   -d --dir         Selects a directory to be used for temporary files written when\n",
      "                    generating report images. Defaults to system temp directory if\n",
      "                    not specified.\n",
      "                    \n",
      "BUGS\n",
      "\n",
      "    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk\n",
      "    or in www.bioinformatics.babraham.ac.uk/bugzilla/\n",
      "                   \n",
      "    \n"
     ]
    }
   ],
   "source": [
    "!./FastQC/fastqc --help"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "10Fs3qrmHmJi",
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "# Выравнивание"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "-z4GQiMfU1wf",
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "## Скачиваем чтения\n",
    "\n",
    "Для примера чтения были взяты из эксперимента https://www.encodeproject.org/experiments/ENCSR000AKC/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "lFWnMg0uU0sJ",
    "outputId": "346b5af7-c5d1-4121-e04d-8d4e96d40dcd",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "--2022-03-07 17:59:45--  https://www.encodeproject.org/files/ENCFF738ERP/@@download/ENCFF738ERP.fastq.gz\n",
      "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
      "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
      "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
      "Location: https://encode-public.s3.amazonaws.com/2016/10/07/2fcb2fac-f190-4e22-b1f6-bf48983c9666/ENCFF738ERP.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF738ERP.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX5WD2ZKUR&Signature=%2BIZnynTD8s3FOR%2FyBPGa%2FFSiDJw%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEEIaCXVzLXdlc3QtMiJHMEUCIAbvj65vd86oXCN2OI%2FSa2zWVLo4xzssXBQJshq7oLeYAiEAt0zJumE%2BSaQeUp5uQcanqBeWIb2GrOC0dEdxQY1V1LUq0gQIGxAAGgwyMjA3NDg3MTQ4NjMiDAel%2FLrLuPTq3ZS4PSqvBCbljA55zEMpTdBH7gJzBZhMITvv3XsatAQpXDsCUryUbAFrLZRtteBBlxYe8d01CHKkx5kyJ%2BL%2FgjpoIFXDk44ugrYgvEnoGJ3wZJ%2FrotZOz7cP8SvZmBXss8U5XHd6gt8rAg49gu3BQgYU3PlRPdcnJZ%2BVj18gQ1GKfVKMzvHbaIUDh%2BbnoM0w8k6m1bPyuCETyiYE%2FmW0PsFLLsyGKcuUw3DykruB5KwPRk1%2FmHa1%2BHo1ih2adT8dgrcicBj8cvzmaj%2FRR87Z20j9P6y7NV7ObOO5bny7%2FB50LQecz31%2BaMgIpiqvLnBfC5V%2BF0G92nKoynYiBR3UrQGNZGmQfDGdORRyCTPRuCMShyFcKYH7JUprygCzV0Ie4ZDLuDqdw41lrUCWXCJgRSRSrHHltZVIVNqF7uxec7QslNBauSVaG7m8nKWrKnDsrw9qHYTpRBoTRhl7XLnF0DjUSzVwWSx%2FG9G6iN9bwM8S%2BSk4mIAUMtwl3rtbwzuiOGCnjVUiHeXmpSvLIQ73THwI7%2B0Qopbw00ZmbYXW%2FA9lv92ty%2B4GjptyxkS%2Bu3nl2KpbYonsfmEc89F%2BLzu8Z79m%2BylocOI3aOpnSr9PZbqaTHuhOhnePSUkudxr1HJWBDuEj6zYoD3sRCFVnHoKNWJ%2By51V4HzvHKA1dkJgJy3TzJgQRhhq2OdFnVSorcfbjiFXPj4BMywXWFWuEcFENmftHIRcA5mOdU34geOyTaRdM78QR9wwzMTqkwY6qQHVTcY5V0unv8pUy%2BdvRGKyoqSouY4%2FRWkBlMCUFl3ilEK6AfK42N6d0TlvCgu%2BMkijh1WZr1p7EVJy2Amm9fwGVb8lOJLeUfMbbQQg2ASSQhOdjBlR0Pm3HUwLncZNVudxo6KTRElahK1UsUygvjLZYQlKBBcaFXVt7OgG7vHspPkxDs6s%2Bc7pn0eB4jmAafmo40trjJ4o%2FcQwgOEEwrMqzi0zV7EFS5Bs&Expires=1652335186 [following]\n",
      "--2022-03-07 17:59:46--  https://encode-public.s3.amazonaws.com/2016/10/07/2fcb2fac-f190-4e22-b1f6-bf48983c9666/ENCFF738ERP.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF738ERP.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX5WD2ZKUR&Signature=%2BIZnynTD8s3FOR%2FyBPGa%2FFSiDJw%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEEIaCXVzLXdlc3QtMiJHMEUCIAbvj65vd86oXCN2OI%2FSa2zWVLo4xzssXBQJshq7oLeYAiEAt0zJumE%2BSaQeUp5uQcanqBeWIb2GrOC0dEdxQY1V1LUq0gQIGxAAGgwyMjA3NDg3MTQ4NjMiDAel%2FLrLuPTq3ZS4PSqvBCbljA55zEMpTdBH7gJzBZhMITvv3XsatAQpXDsCUryUbAFrLZRtteBBlxYe8d01CHKkx5kyJ%2BL%2FgjpoIFXDk44ugrYgvEnoGJ3wZJ%2FrotZOz7cP8SvZmBXss8U5XHd6gt8rAg49gu3BQgYU3PlRPdcnJZ%2BVj18gQ1GKfVKMzvHbaIUDh%2BbnoM0w8k6m1bPyuCETyiYE%2FmW0PsFLLsyGKcuUw3DykruB5KwPRk1%2FmHa1%2BHo1ih2adT8dgrcicBj8cvzmaj%2FRR87Z20j9P6y7NV7ObOO5bny7%2FB50LQecz31%2BaMgIpiqvLnBfC5V%2BF0G92nKoynYiBR3UrQGNZGmQfDGdORRyCTPRuCMShyFcKYH7JUprygCzV0Ie4ZDLuDqdw41lrUCWXCJgRSRSrHHltZVIVNqF7uxec7QslNBauSVaG7m8nKWrKnDsrw9qHYTpRBoTRhl7XLnF0DjUSzVwWSx%2FG9G6iN9bwM8S%2BSk4mIAUMtwl3rtbwzuiOGCnjVUiHeXmpSvLIQ73THwI7%2B0Qopbw00ZmbYXW%2FA9lv92ty%2B4GjptyxkS%2Bu3nl2KpbYonsfmEc89F%2BLzu8Z79m%2BylocOI3aOpnSr9PZbqaTHuhOhnePSUkudxr1HJWBDuEj6zYoD3sRCFVnHoKNWJ%2By51V4HzvHKA1dkJgJy3TzJgQRhhq2OdFnVSorcfbjiFXPj4BMywXWFWuEcFENmftHIRcA5mOdU34geOyTaRdM78QR9wwzMTqkwY6qQHVTcY5V0unv8pUy%2BdvRGKyoqSouY4%2FRWkBlMCUFl3ilEK6AfK42N6d0TlvCgu%2BMkijh1WZr1p7EVJy2Amm9fwGVb8lOJLeUfMbbQQg2ASSQhOdjBlR0Pm3HUwLncZNVudxo6KTRElahK1UsUygvjLZYQlKBBcaFXVt7OgG7vHspPkxDs6s%2Bc7pn0eB4jmAafmo40trjJ4o%2FcQwgOEEwrMqzi0zV7EFS5Bs&Expires=1652335186\n",
      "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.216.178\n",
      "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.216.178|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 1282350894 (1.2G) [binary/octet-stream]\n",
      "Saving to: ‘ENCFF738ERP.fastq.gz’\n",
      "\n",
      "ENCFF738ERP.fastq.g 100%[===================>]   1.19G  25.9MB/s    in 45s     \n",
      "\n",
      "2022-03-07 18:00:31 (27.1 MB/s) - ‘ENCFF738ERP.fastq.gz’ saved [1282350894/1282350894]\n",
      "\n",
      "--2022-03-07 18:00:31--  https://www.encodeproject.org/files/ENCFF134NCG/@@download/ENCFF134NCG.fastq.gz\n",
      "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
      "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
      "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
      "Location: https://encode-public.s3.amazonaws.com/2016/10/07/d19cd45e-06d1-4680-9551-540bf5b871b7/ENCFF134NCG.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF134NCG.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX5WD2ZKUR&Signature=uaH7OThf%2BBxn6O%2F40knlgYhK6ik%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEEIaCXVzLXdlc3QtMiJHMEUCIAbvj65vd86oXCN2OI%2FSa2zWVLo4xzssXBQJshq7oLeYAiEAt0zJumE%2BSaQeUp5uQcanqBeWIb2GrOC0dEdxQY1V1LUq0gQIGxAAGgwyMjA3NDg3MTQ4NjMiDAel%2FLrLuPTq3ZS4PSqvBCbljA55zEMpTdBH7gJzBZhMITvv3XsatAQpXDsCUryUbAFrLZRtteBBlxYe8d01CHKkx5kyJ%2BL%2FgjpoIFXDk44ugrYgvEnoGJ3wZJ%2FrotZOz7cP8SvZmBXss8U5XHd6gt8rAg49gu3BQgYU3PlRPdcnJZ%2BVj18gQ1GKfVKMzvHbaIUDh%2BbnoM0w8k6m1bPyuCETyiYE%2FmW0PsFLLsyGKcuUw3DykruB5KwPRk1%2FmHa1%2BHo1ih2adT8dgrcicBj8cvzmaj%2FRR87Z20j9P6y7NV7ObOO5bny7%2FB50LQecz31%2BaMgIpiqvLnBfC5V%2BF0G92nKoynYiBR3UrQGNZGmQfDGdORRyCTPRuCMShyFcKYH7JUprygCzV0Ie4ZDLuDqdw41lrUCWXCJgRSRSrHHltZVIVNqF7uxec7QslNBauSVaG7m8nKWrKnDsrw9qHYTpRBoTRhl7XLnF0DjUSzVwWSx%2FG9G6iN9bwM8S%2BSk4mIAUMtwl3rtbwzuiOGCnjVUiHeXmpSvLIQ73THwI7%2B0Qopbw00ZmbYXW%2FA9lv92ty%2B4GjptyxkS%2Bu3nl2KpbYonsfmEc89F%2BLzu8Z79m%2BylocOI3aOpnSr9PZbqaTHuhOhnePSUkudxr1HJWBDuEj6zYoD3sRCFVnHoKNWJ%2By51V4HzvHKA1dkJgJy3TzJgQRhhq2OdFnVSorcfbjiFXPj4BMywXWFWuEcFENmftHIRcA5mOdU34geOyTaRdM78QR9wwzMTqkwY6qQHVTcY5V0unv8pUy%2BdvRGKyoqSouY4%2FRWkBlMCUFl3ilEK6AfK42N6d0TlvCgu%2BMkijh1WZr1p7EVJy2Amm9fwGVb8lOJLeUfMbbQQg2ASSQhOdjBlR0Pm3HUwLncZNVudxo6KTRElahK1UsUygvjLZYQlKBBcaFXVt7OgG7vHspPkxDs6s%2Bc7pn0eB4jmAafmo40trjJ4o%2FcQwgOEEwrMqzi0zV7EFS5Bs&Expires=1652335232 [following]\n",
      "--2022-03-07 18:00:32--  https://encode-public.s3.amazonaws.com/2016/10/07/d19cd45e-06d1-4680-9551-540bf5b871b7/ENCFF134NCG.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF134NCG.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX5WD2ZKUR&Signature=uaH7OThf%2BBxn6O%2F40knlgYhK6ik%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEEIaCXVzLXdlc3QtMiJHMEUCIAbvj65vd86oXCN2OI%2FSa2zWVLo4xzssXBQJshq7oLeYAiEAt0zJumE%2BSaQeUp5uQcanqBeWIb2GrOC0dEdxQY1V1LUq0gQIGxAAGgwyMjA3NDg3MTQ4NjMiDAel%2FLrLuPTq3ZS4PSqvBCbljA55zEMpTdBH7gJzBZhMITvv3XsatAQpXDsCUryUbAFrLZRtteBBlxYe8d01CHKkx5kyJ%2BL%2FgjpoIFXDk44ugrYgvEnoGJ3wZJ%2FrotZOz7cP8SvZmBXss8U5XHd6gt8rAg49gu3BQgYU3PlRPdcnJZ%2BVj18gQ1GKfVKMzvHbaIUDh%2BbnoM0w8k6m1bPyuCETyiYE%2FmW0PsFLLsyGKcuUw3DykruB5KwPRk1%2FmHa1%2BHo1ih2adT8dgrcicBj8cvzmaj%2FRR87Z20j9P6y7NV7ObOO5bny7%2FB50LQecz31%2BaMgIpiqvLnBfC5V%2BF0G92nKoynYiBR3UrQGNZGmQfDGdORRyCTPRuCMShyFcKYH7JUprygCzV0Ie4ZDLuDqdw41lrUCWXCJgRSRSrHHltZVIVNqF7uxec7QslNBauSVaG7m8nKWrKnDsrw9qHYTpRBoTRhl7XLnF0DjUSzVwWSx%2FG9G6iN9bwM8S%2BSk4mIAUMtwl3rtbwzuiOGCnjVUiHeXmpSvLIQ73THwI7%2B0Qopbw00ZmbYXW%2FA9lv92ty%2B4GjptyxkS%2Bu3nl2KpbYonsfmEc89F%2BLzu8Z79m%2BylocOI3aOpnSr9PZbqaTHuhOhnePSUkudxr1HJWBDuEj6zYoD3sRCFVnHoKNWJ%2By51V4HzvHKA1dkJgJy3TzJgQRhhq2OdFnVSorcfbjiFXPj4BMywXWFWuEcFENmftHIRcA5mOdU34geOyTaRdM78QR9wwzMTqkwY6qQHVTcY5V0unv8pUy%2BdvRGKyoqSouY4%2FRWkBlMCUFl3ilEK6AfK42N6d0TlvCgu%2BMkijh1WZr1p7EVJy2Amm9fwGVb8lOJLeUfMbbQQg2ASSQhOdjBlR0Pm3HUwLncZNVudxo6KTRElahK1UsUygvjLZYQlKBBcaFXVt7OgG7vHspPkxDs6s%2Bc7pn0eB4jmAafmo40trjJ4o%2FcQwgOEEwrMqzi0zV7EFS5Bs&Expires=1652335232\n",
      "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.228.115\n",
      "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.228.115|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 1394349171 (1.3G) [binary/octet-stream]\n",
      "Saving to: ‘ENCFF134NCG.fastq.gz’\n",
      "\n",
      "ENCFF134NCG.fastq.g 100%[===================>]   1.30G  31.5MB/s    in 41s     \n",
      "\n",
      "2022-03-07 18:01:13 (32.4 MB/s) - ‘ENCFF134NCG.fastq.gz’ saved [1394349171/1394349171]\n",
      "\n",
      "--2022-03-07 18:01:13--  https://www.encodeproject.org/files/ENCFF222VRH/@@download/ENCFF222VRH.fastq.gz\n",
      "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
      "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
      "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
      "Location: https://encode-public.s3.amazonaws.com/2016/10/08/7a0fd3ba-55fb-4558-9dd4-45c393426ab7/ENCFF222VRH.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF222VRH.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX5WD2ZKUR&Signature=EvKve1pzoQRNBq4%2F0riHHmkHMOo%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEEIaCXVzLXdlc3QtMiJHMEUCIAbvj65vd86oXCN2OI%2FSa2zWVLo4xzssXBQJshq7oLeYAiEAt0zJumE%2BSaQeUp5uQcanqBeWIb2GrOC0dEdxQY1V1LUq0gQIGxAAGgwyMjA3NDg3MTQ4NjMiDAel%2FLrLuPTq3ZS4PSqvBCbljA55zEMpTdBH7gJzBZhMITvv3XsatAQpXDsCUryUbAFrLZRtteBBlxYe8d01CHKkx5kyJ%2BL%2FgjpoIFXDk44ugrYgvEnoGJ3wZJ%2FrotZOz7cP8SvZmBXss8U5XHd6gt8rAg49gu3BQgYU3PlRPdcnJZ%2BVj18gQ1GKfVKMzvHbaIUDh%2BbnoM0w8k6m1bPyuCETyiYE%2FmW0PsFLLsyGKcuUw3DykruB5KwPRk1%2FmHa1%2BHo1ih2adT8dgrcicBj8cvzmaj%2FRR87Z20j9P6y7NV7ObOO5bny7%2FB50LQecz31%2BaMgIpiqvLnBfC5V%2BF0G92nKoynYiBR3UrQGNZGmQfDGdORRyCTPRuCMShyFcKYH7JUprygCzV0Ie4ZDLuDqdw41lrUCWXCJgRSRSrHHltZVIVNqF7uxec7QslNBauSVaG7m8nKWrKnDsrw9qHYTpRBoTRhl7XLnF0DjUSzVwWSx%2FG9G6iN9bwM8S%2BSk4mIAUMtwl3rtbwzuiOGCnjVUiHeXmpSvLIQ73THwI7%2B0Qopbw00ZmbYXW%2FA9lv92ty%2B4GjptyxkS%2Bu3nl2KpbYonsfmEc89F%2BLzu8Z79m%2BylocOI3aOpnSr9PZbqaTHuhOhnePSUkudxr1HJWBDuEj6zYoD3sRCFVnHoKNWJ%2By51V4HzvHKA1dkJgJy3TzJgQRhhq2OdFnVSorcfbjiFXPj4BMywXWFWuEcFENmftHIRcA5mOdU34geOyTaRdM78QR9wwzMTqkwY6qQHVTcY5V0unv8pUy%2BdvRGKyoqSouY4%2FRWkBlMCUFl3ilEK6AfK42N6d0TlvCgu%2BMkijh1WZr1p7EVJy2Amm9fwGVb8lOJLeUfMbbQQg2ASSQhOdjBlR0Pm3HUwLncZNVudxo6KTRElahK1UsUygvjLZYQlKBBcaFXVt7OgG7vHspPkxDs6s%2Bc7pn0eB4jmAafmo40trjJ4o%2FcQwgOEEwrMqzi0zV7EFS5Bs&Expires=1652335274 [following]\n",
      "--2022-03-07 18:01:14--  https://encode-public.s3.amazonaws.com/2016/10/08/7a0fd3ba-55fb-4558-9dd4-45c393426ab7/ENCFF222VRH.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF222VRH.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX5WD2ZKUR&Signature=EvKve1pzoQRNBq4%2F0riHHmkHMOo%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEEIaCXVzLXdlc3QtMiJHMEUCIAbvj65vd86oXCN2OI%2FSa2zWVLo4xzssXBQJshq7oLeYAiEAt0zJumE%2BSaQeUp5uQcanqBeWIb2GrOC0dEdxQY1V1LUq0gQIGxAAGgwyMjA3NDg3MTQ4NjMiDAel%2FLrLuPTq3ZS4PSqvBCbljA55zEMpTdBH7gJzBZhMITvv3XsatAQpXDsCUryUbAFrLZRtteBBlxYe8d01CHKkx5kyJ%2BL%2FgjpoIFXDk44ugrYgvEnoGJ3wZJ%2FrotZOz7cP8SvZmBXss8U5XHd6gt8rAg49gu3BQgYU3PlRPdcnJZ%2BVj18gQ1GKfVKMzvHbaIUDh%2BbnoM0w8k6m1bPyuCETyiYE%2FmW0PsFLLsyGKcuUw3DykruB5KwPRk1%2FmHa1%2BHo1ih2adT8dgrcicBj8cvzmaj%2FRR87Z20j9P6y7NV7ObOO5bny7%2FB50LQecz31%2BaMgIpiqvLnBfC5V%2BF0G92nKoynYiBR3UrQGNZGmQfDGdORRyCTPRuCMShyFcKYH7JUprygCzV0Ie4ZDLuDqdw41lrUCWXCJgRSRSrHHltZVIVNqF7uxec7QslNBauSVaG7m8nKWrKnDsrw9qHYTpRBoTRhl7XLnF0DjUSzVwWSx%2FG9G6iN9bwM8S%2BSk4mIAUMtwl3rtbwzuiOGCnjVUiHeXmpSvLIQ73THwI7%2B0Qopbw00ZmbYXW%2FA9lv92ty%2B4GjptyxkS%2Bu3nl2KpbYonsfmEc89F%2BLzu8Z79m%2BylocOI3aOpnSr9PZbqaTHuhOhnePSUkudxr1HJWBDuEj6zYoD3sRCFVnHoKNWJ%2By51V4HzvHKA1dkJgJy3TzJgQRhhq2OdFnVSorcfbjiFXPj4BMywXWFWuEcFENmftHIRcA5mOdU34geOyTaRdM78QR9wwzMTqkwY6qQHVTcY5V0unv8pUy%2BdvRGKyoqSouY4%2FRWkBlMCUFl3ilEK6AfK42N6d0TlvCgu%2BMkijh1WZr1p7EVJy2Amm9fwGVb8lOJLeUfMbbQQg2ASSQhOdjBlR0Pm3HUwLncZNVudxo6KTRElahK1UsUygvjLZYQlKBBcaFXVt7OgG7vHspPkxDs6s%2Bc7pn0eB4jmAafmo40trjJ4o%2FcQwgOEEwrMqzi0zV7EFS5Bs&Expires=1652335274\n",
      "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.92.146.241\n",
      "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.92.146.241|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 3616544950 (3.4G) [binary/octet-stream]\n",
      "Saving to: ‘ENCFF222VRH.fastq.gz’\n",
      "\n",
      "ENCFF222VRH.fastq.g 100%[===================>]   3.37G  30.1MB/s    in 1m 42s  \n",
      "\n",
      "2022-03-07 18:02:56 (33.9 MB/s) - ‘ENCFF222VRH.fastq.gz’ saved [3616544950/3616544950]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# ChIP-seq на гистоновой метке\n",
    "!wget https://www.encodeproject.org/files/ENCFF738ERP/@@download/ENCFF738ERP.fastq.gz\n",
    "!wget https://www.encodeproject.org/files/ENCFF134NCG/@@download/ENCFF134NCG.fastq.gz\n",
    "# ChIP-seq контроль\n",
    "!wget https://www.encodeproject.org/files/ENCFF222VRH/@@download/ENCFF222VRH.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "id": "v73fH7WiVDu9",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "# распаковка ахрива\n",
    "!gzip -d /content/ENCFF738ERP.fastq.gz /content/ENCFF134NCG.fastq.gz /content/ENCFF222VRH.fastq.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "6sPcIefNHx-L",
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "## FastQC (или multiQC может быть использован в работе)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "5XM6jBcuGjMV",
    "outputId": "9c138bd8-730d-4abe-de4d-fe8403dfbe08",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "Started analysis of ENCFF738ERP.fastq\n",
      "Approx 5% complete for ENCFF738ERP.fastq\n",
      "Approx 10% complete for ENCFF738ERP.fastq\n",
      "Approx 15% complete for ENCFF738ERP.fastq\n",
      "Approx 20% complete for ENCFF738ERP.fastq\n",
      "Approx 25% complete for ENCFF738ERP.fastq\n",
      "Approx 30% complete for ENCFF738ERP.fastq\n",
      "Approx 35% complete for ENCFF738ERP.fastq\n",
      "Approx 40% complete for ENCFF738ERP.fastq\n",
      "Approx 45% complete for ENCFF738ERP.fastq\n",
      "Approx 50% complete for ENCFF738ERP.fastq\n",
      "Approx 55% complete for ENCFF738ERP.fastq\n",
      "Approx 60% complete for ENCFF738ERP.fastq\n",
      "Approx 65% complete for ENCFF738ERP.fastq\n",
      "Approx 70% complete for ENCFF738ERP.fastq\n",
      "Approx 75% complete for ENCFF738ERP.fastq\n",
      "Approx 80% complete for ENCFF738ERP.fastq\n",
      "Approx 85% complete for ENCFF738ERP.fastq\n",
      "Approx 90% complete for ENCFF738ERP.fastq\n",
      "Approx 95% complete for ENCFF738ERP.fastq\n",
      "Analysis complete for ENCFF738ERP.fastq\n",
      "Started analysis of ENCFF134NCG.fastq\n",
      "Approx 5% complete for ENCFF134NCG.fastq\n",
      "Approx 10% complete for ENCFF134NCG.fastq\n",
      "Approx 15% complete for ENCFF134NCG.fastq\n",
      "Approx 20% complete for ENCFF134NCG.fastq\n",
      "Approx 25% complete for ENCFF134NCG.fastq\n",
      "Approx 30% complete for ENCFF134NCG.fastq\n",
      "Approx 35% complete for ENCFF134NCG.fastq\n",
      "Approx 40% complete for ENCFF134NCG.fastq\n",
      "Approx 45% complete for ENCFF134NCG.fastq\n",
      "Approx 50% complete for ENCFF134NCG.fastq\n",
      "Approx 55% complete for ENCFF134NCG.fastq\n",
      "Approx 60% complete for ENCFF134NCG.fastq\n",
      "Approx 65% complete for ENCFF134NCG.fastq\n",
      "Approx 70% complete for ENCFF134NCG.fastq\n",
      "Approx 75% complete for ENCFF134NCG.fastq\n",
      "Approx 80% complete for ENCFF134NCG.fastq\n",
      "Approx 85% complete for ENCFF134NCG.fastq\n",
      "Approx 90% complete for ENCFF134NCG.fastq\n",
      "Approx 95% complete for ENCFF134NCG.fastq\n",
      "Analysis complete for ENCFF134NCG.fastq\n",
      "Started analysis of ENCFF222VRH.fastq\n",
      "Approx 5% complete for ENCFF222VRH.fastq\n",
      "Approx 10% complete for ENCFF222VRH.fastq\n",
      "Approx 15% complete for ENCFF222VRH.fastq\n",
      "Approx 20% complete for ENCFF222VRH.fastq\n",
      "Approx 25% complete for ENCFF222VRH.fastq\n",
      "Approx 30% complete for ENCFF222VRH.fastq\n",
      "Approx 35% complete for ENCFF222VRH.fastq\n",
      "Approx 40% complete for ENCFF222VRH.fastq\n",
      "Approx 45% complete for ENCFF222VRH.fastq\n",
      "Approx 50% complete for ENCFF222VRH.fastq\n",
      "Approx 55% complete for ENCFF222VRH.fastq\n",
      "Approx 60% complete for ENCFF222VRH.fastq\n",
      "Approx 65% complete for ENCFF222VRH.fastq\n",
      "Approx 70% complete for ENCFF222VRH.fastq\n",
      "Approx 75% complete for ENCFF222VRH.fastq\n",
      "Approx 80% complete for ENCFF222VRH.fastq\n",
      "Approx 85% complete for ENCFF222VRH.fastq\n",
      "Approx 90% complete for ENCFF222VRH.fastq\n",
      "Approx 95% complete for ENCFF222VRH.fastq\n",
      "Analysis complete for ENCFF222VRH.fastq\n"
     ]
    }
   ],
   "source": [
    "!./FastQC/fastqc /content/ENCFF738ERP.fastq\n",
    "!./FastQC/fastqc /content/ENCFF134NCG.fastq\n",
    "!./FastQC/fastqc /content/ENCFF222VRH.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "Jnc4H1xYeFTp",
    "outputId": "7016e852-1137-4a83-db35-cabe3c588d66",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
      "Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /usr/local\n",
      "\n",
      "  added / updated specs:\n",
      "    - multiqc\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    click-8.1.3                |   py37h89c1867_0         145 KB  conda-forge\n",
      "    colorama-0.4.4             |     pyh9f0ad1d_0          18 KB  conda-forge\n",
      "    coloredlogs-15.0.1         |     pyhd8ed1ab_3          40 KB  conda-forge\n",
      "    colormath-3.0.0            |             py_2          34 KB  conda-forge\n",
      "    commonmark-0.9.1           |             py_0          46 KB  conda-forge\n",
      "    cycler-0.11.0              |     pyhd8ed1ab_0          10 KB  conda-forge\n",
      "    dataclasses-0.8            |     pyhc8e2a94_3          10 KB  conda-forge\n",
      "    decorator-4.4.2            |             py_0          11 KB  conda-forge\n",
      "    freetype-2.10.4            |       h0708190_1         890 KB  conda-forge\n",
      "    future-0.18.2              |   py37h89c1867_5         713 KB  conda-forge\n",
      "    humanfriendly-10.0         |   py37h89c1867_2         118 KB  conda-forge\n",
      "    importlib-metadata-4.11.3  |   py37h89c1867_1          33 KB  conda-forge\n",
      "    jinja2-3.1.2               |     pyhd8ed1ab_0          99 KB  conda-forge\n",
      "    jpeg-9e                    |       h166bdaf_1         268 KB  conda-forge\n",
      "    kiwisolver-1.4.2           |   py37h7cecad7_1          73 KB  conda-forge\n",
      "    libblas-3.9.0              |14_linux64_openblas          12 KB  conda-forge\n",
      "    libcblas-3.9.0             |14_linux64_openblas          12 KB  conda-forge\n",
      "    libgcc-ng-11.2.0           |      h1d223b6_16         902 KB  conda-forge\n",
      "    libgfortran-ng-11.2.0      |      h69a702a_16          23 KB  conda-forge\n",
      "    libgfortran5-11.2.0        |      h5c6108e_16         1.7 MB  conda-forge\n",
      "    libgomp-11.2.0             |      h1d223b6_16         428 KB  conda-forge\n",
      "    liblapack-3.9.0            |14_linux64_openblas          12 KB  conda-forge\n",
      "    libopenblas-0.3.20         |pthreads_h78a6416_0        10.1 MB  conda-forge\n",
      "    libpng-1.6.37              |       h21135ba_2         306 KB  conda-forge\n",
      "    libstdcxx-ng-11.2.0        |      he4da1e4_16         4.2 MB  conda-forge\n",
      "    libtiff-4.0.10             |    hc3755c2_1005         602 KB  conda-forge\n",
      "    lzstring-1.0.4             |          py_1001           6 KB  conda-forge\n",
      "    markdown-3.3.7             |     pyhd8ed1ab_0          67 KB  conda-forge\n",
      "    markupsafe-2.1.1           |   py37h540881e_1          22 KB  conda-forge\n",
      "    matplotlib-base-3.4.3      |   py37h1058ff1_0         7.2 MB  conda-forge\n",
      "    multiqc-1.12               |     pyhdfd78af_0         942 KB  bioconda\n",
      "    networkx-2.5.1             |     pyhd8ed1ab_0         1.2 MB  conda-forge\n",
      "    numpy-1.21.6               |   py37h976b520_0         6.1 MB  conda-forge\n",
      "    olefile-0.46               |     pyh9f0ad1d_1          32 KB  conda-forge\n",
      "    openssl-1.1.1o             |       h166bdaf_0         2.1 MB  conda-forge\n",
      "    pillow-6.2.1               |   py37h6b7be26_0         637 KB  conda-forge\n",
      "    pygments-2.12.0            |     pyhd8ed1ab_0         817 KB  conda-forge\n",
      "    pyparsing-3.0.8            |     pyhd8ed1ab_0          79 KB  conda-forge\n",
      "    python-dateutil-2.8.2      |     pyhd8ed1ab_0         240 KB  conda-forge\n",
      "    pyyaml-6.0                 |   py37h540881e_4         178 KB  conda-forge\n",
      "    rich-12.4.1                |     pyhd8ed1ab_0         164 KB  conda-forge\n",
      "    simplejson-3.17.6          |   py37h540881e_1         103 KB  conda-forge\n",
      "    spectra-0.0.11             |             py_1          19 KB  conda-forge\n",
      "    tornado-6.1                |   py37h540881e_3         646 KB  conda-forge\n",
      "    typing-extensions-4.2.0    |       hd8ed1ab_1           8 KB  conda-forge\n",
      "    typing_extensions-4.2.0    |     pyha770c72_1          27 KB  conda-forge\n",
      "    zipp-3.8.0                 |     pyhd8ed1ab_0          12 KB  conda-forge\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:        41.3 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  click              conda-forge/linux-64::click-8.1.3-py37h89c1867_0\n",
      "  colorama           conda-forge/noarch::colorama-0.4.4-pyh9f0ad1d_0\n",
      "  coloredlogs        conda-forge/noarch::coloredlogs-15.0.1-pyhd8ed1ab_3\n",
      "  colormath          conda-forge/noarch::colormath-3.0.0-py_2\n",
      "  commonmark         conda-forge/noarch::commonmark-0.9.1-py_0\n",
      "  cycler             conda-forge/noarch::cycler-0.11.0-pyhd8ed1ab_0\n",
      "  dataclasses        conda-forge/noarch::dataclasses-0.8-pyhc8e2a94_3\n",
      "  decorator          conda-forge/noarch::decorator-4.4.2-py_0\n",
      "  freetype           conda-forge/linux-64::freetype-2.10.4-h0708190_1\n",
      "  future             conda-forge/linux-64::future-0.18.2-py37h89c1867_5\n",
      "  humanfriendly      conda-forge/linux-64::humanfriendly-10.0-py37h89c1867_2\n",
      "  importlib-metadata conda-forge/linux-64::importlib-metadata-4.11.3-py37h89c1867_1\n",
      "  jinja2             conda-forge/noarch::jinja2-3.1.2-pyhd8ed1ab_0\n",
      "  jpeg               conda-forge/linux-64::jpeg-9e-h166bdaf_1\n",
      "  kiwisolver         conda-forge/linux-64::kiwisolver-1.4.2-py37h7cecad7_1\n",
      "  libblas            conda-forge/linux-64::libblas-3.9.0-14_linux64_openblas\n",
      "  libcblas           conda-forge/linux-64::libcblas-3.9.0-14_linux64_openblas\n",
      "  libgfortran-ng     conda-forge/linux-64::libgfortran-ng-11.2.0-h69a702a_16\n",
      "  libgfortran5       conda-forge/linux-64::libgfortran5-11.2.0-h5c6108e_16\n",
      "  liblapack          conda-forge/linux-64::liblapack-3.9.0-14_linux64_openblas\n",
      "  libopenblas        conda-forge/linux-64::libopenblas-0.3.20-pthreads_h78a6416_0\n",
      "  libpng             conda-forge/linux-64::libpng-1.6.37-h21135ba_2\n",
      "  libtiff            conda-forge/linux-64::libtiff-4.0.10-hc3755c2_1005\n",
      "  lzstring           conda-forge/noarch::lzstring-1.0.4-py_1001\n",
      "  markdown           conda-forge/noarch::markdown-3.3.7-pyhd8ed1ab_0\n",
      "  markupsafe         conda-forge/linux-64::markupsafe-2.1.1-py37h540881e_1\n",
      "  matplotlib-base    conda-forge/linux-64::matplotlib-base-3.4.3-py37h1058ff1_0\n",
      "  multiqc            bioconda/noarch::multiqc-1.12-pyhdfd78af_0\n",
      "  networkx           conda-forge/noarch::networkx-2.5.1-pyhd8ed1ab_0\n",
      "  numpy              conda-forge/linux-64::numpy-1.21.6-py37h976b520_0\n",
      "  olefile            conda-forge/noarch::olefile-0.46-pyh9f0ad1d_1\n",
      "  pillow             conda-forge/linux-64::pillow-6.2.1-py37h6b7be26_0\n",
      "  pygments           conda-forge/noarch::pygments-2.12.0-pyhd8ed1ab_0\n",
      "  pyparsing          conda-forge/noarch::pyparsing-3.0.8-pyhd8ed1ab_0\n",
      "  python-dateutil    conda-forge/noarch::python-dateutil-2.8.2-pyhd8ed1ab_0\n",
      "  pyyaml             conda-forge/linux-64::pyyaml-6.0-py37h540881e_4\n",
      "  rich               conda-forge/noarch::rich-12.4.1-pyhd8ed1ab_0\n",
      "  simplejson         conda-forge/linux-64::simplejson-3.17.6-py37h540881e_1\n",
      "  spectra            conda-forge/noarch::spectra-0.0.11-py_1\n",
      "  tornado            conda-forge/linux-64::tornado-6.1-py37h540881e_3\n",
      "  typing-extensions  conda-forge/noarch::typing-extensions-4.2.0-hd8ed1ab_1\n",
      "  typing_extensions  conda-forge/noarch::typing_extensions-4.2.0-pyha770c72_1\n",
      "  zipp               conda-forge/noarch::zipp-3.8.0-pyhd8ed1ab_0\n",
      "\n",
      "The following packages will be UPDATED:\n",
      "\n",
      "  libgcc-ng                               9.3.0-h2828fa1_18 --> 11.2.0-h1d223b6_16\n",
      "  libgomp                                 9.3.0-h2828fa1_18 --> 11.2.0-h1d223b6_16\n",
      "  libstdcxx-ng                            9.3.0-h6de172a_18 --> 11.2.0-he4da1e4_16\n",
      "  openssl                                 1.1.1j-h7f98852_0 --> 1.1.1o-h166bdaf_0\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "colorama-0.4.4       | 18 KB     | : 100% 1.0/1 [00:00<00:00,  7.89it/s]               \n",
      "future-0.18.2        | 713 KB    | : 100% 1.0/1 [00:00<00:00,  3.19it/s]\n",
      "libblas-3.9.0        | 12 KB     | : 100% 1.0/1 [00:00<00:00, 20.72it/s]\n",
      "typing-extensions-4. | 8 KB      | : 100% 1.0/1 [00:00<00:00, 21.90it/s]\n",
      "jpeg-9e              | 268 KB    | : 100% 1.0/1 [00:00<00:00,  8.67it/s]\n",
      "decorator-4.4.2      | 11 KB     | : 100% 1.0/1 [00:00<00:00, 14.55it/s]\n",
      "pyyaml-6.0           | 178 KB    | : 100% 1.0/1 [00:00<00:00, 11.70it/s]\n",
      "libgfortran-ng-11.2. | 23 KB     | : 100% 1.0/1 [00:00<00:00, 14.86it/s]\n",
      "freetype-2.10.4      | 890 KB    | : 100% 1.0/1 [00:00<00:00,  4.31it/s]\n",
      "libcblas-3.9.0       | 12 KB     | : 100% 1.0/1 [00:00<00:00, 21.56it/s]\n",
      "libgfortran5-11.2.0  | 1.7 MB    | : 100% 1.0/1 [00:00<00:00,  2.20it/s]\n",
      "python-dateutil-2.8. | 240 KB    | : 100% 1.0/1 [00:00<00:00, 10.50it/s]\n",
      "jinja2-3.1.2         | 99 KB     | : 100% 1.0/1 [00:00<00:00, 14.00it/s]\n",
      "lzstring-1.0.4       | 6 KB      | : 100% 1.0/1 [00:00<00:00, 16.80it/s]\n",
      "importlib-metadata-4 | 33 KB     | : 100% 1.0/1 [00:00<00:00, 20.68it/s]\n",
      "multiqc-1.12         | 942 KB    | : 100% 1.0/1 [00:00<00:00,  2.30it/s]\n",
      "libstdcxx-ng-11.2.0  | 4.2 MB    | : 100% 1.0/1 [00:00<00:00,  1.18it/s]\n",
      "libgcc-ng-11.2.0     | 902 KB    | : 100% 1.0/1 [00:00<00:00,  4.83it/s]\n",
      "cycler-0.11.0        | 10 KB     | : 100% 1.0/1 [00:00<00:00, 18.47it/s]\n",
      "rich-12.4.1          | 164 KB    | : 100% 1.0/1 [00:00<00:00,  9.21it/s]\n",
      "commonmark-0.9.1     | 46 KB     | : 100% 1.0/1 [00:00<00:00, 19.56it/s]\n",
      "click-8.1.3          | 145 KB    | : 100% 1.0/1 [00:00<00:00, 12.53it/s]\n",
      "coloredlogs-15.0.1   | 40 KB     | : 100% 1.0/1 [00:00<00:00, 15.37it/s]\n",
      "markupsafe-2.1.1     | 22 KB     | : 100% 1.0/1 [00:00<00:00, 17.77it/s]\n",
      "zipp-3.8.0           | 12 KB     | : 100% 1.0/1 [00:00<00:00, 22.31it/s]\n",
      "markdown-3.3.7       | 67 KB     | : 100% 1.0/1 [00:00<00:00, 13.94it/s]\n",
      "humanfriendly-10.0   | 118 KB    | : 100% 1.0/1 [00:00<00:00,  9.44it/s]\n",
      "tornado-6.1          | 646 KB    | : 100% 1.0/1 [00:00<00:00,  3.90it/s]\n",
      "colormath-3.0.0      | 34 KB     | : 100% 1.0/1 [00:00<00:00, 17.09it/s]\n",
      "simplejson-3.17.6    | 103 KB    | : 100% 1.0/1 [00:00<00:00, 13.13it/s]\n",
      "libgomp-11.2.0       | 428 KB    | : 100% 1.0/1 [00:00<00:00,  7.58it/s]\n",
      "networkx-2.5.1       | 1.2 MB    | : 100% 1.0/1 [00:00<00:00,  2.16it/s]\n",
      "typing_extensions-4. | 27 KB     | : 100% 1.0/1 [00:00<00:00, 17.30it/s]\n",
      "matplotlib-base-3.4. | 7.2 MB    | : 100% 1.0/1 [00:01<00:00,  1.79s/it]\n",
      "dataclasses-0.8      | 10 KB     | : 100% 1.0/1 [00:00<00:00, 18.33it/s]\n",
      "numpy-1.21.6         | 6.1 MB    | : 100% 1.0/1 [00:01<00:00,  1.70s/it]\n",
      "spectra-0.0.11       | 19 KB     | : 100% 1.0/1 [00:00<00:00, 14.73it/s]\n",
      "pillow-6.2.1         | 637 KB    | : 100% 1.0/1 [00:00<00:00,  5.08it/s]\n",
      "openssl-1.1.1o       | 2.1 MB    | : 100% 1.0/1 [00:00<00:00,  2.09it/s]\n",
      "libtiff-4.0.10       | 602 KB    | : 100% 1.0/1 [00:00<00:00,  5.58it/s]\n",
      "kiwisolver-1.4.2     | 73 KB     | : 100% 1.0/1 [00:00<00:00, 11.89it/s]\n",
      "olefile-0.46         | 32 KB     | : 100% 1.0/1 [00:00<00:00, 19.04it/s]\n",
      "liblapack-3.9.0      | 12 KB     | : 100% 1.0/1 [00:00<00:00, 23.96it/s]\n",
      "pyparsing-3.0.8      | 79 KB     | : 100% 1.0/1 [00:00<00:00, 16.15it/s]\n",
      "libopenblas-0.3.20   | 10.1 MB   | : 100% 1.0/1 [00:02<00:00,  2.28s/it]\n",
      "libpng-1.6.37        | 306 KB    | : 100% 1.0/1 [00:00<00:00,  9.13it/s]\n",
      "pygments-2.12.0      | 817 KB    | : 100% 1.0/1 [00:00<00:00,  3.77it/s]\n",
      "Preparing transaction: \\ \b\b| \b\b/ \b\bdone\n",
      "Verifying transaction: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
      "Executing transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n"
     ]
    }
   ],
   "source": [
    "!conda install -c bioconda -c conda-forge multiqc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "eoGFhVHr5nN3",
    "outputId": "ad1ab68d-20ef-4081-962b-252bdd8e4969",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "\n",
      "  \u001B[34m/\u001B[0m\u001B[32m/\u001B[0m\u001B[31m/\u001B[0m \u001B]8;id=229632;https://multiqc.info\u001B\\\u001B[1mMultiQC\u001B[0m\u001B]8;;\u001B\\ 🔍 \u001B[2m| v1.12\u001B[0m\n",
      "\n",
      "\u001B[34m|           multiqc\u001B[0m | Search path : /content\n",
      "\u001B[2K\u001B[34m|\u001B[0m         \u001B[34msearching\u001B[0m | \u001B[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001B[0m \u001B[35m100%\u001B[0m \u001B[32m286/286\u001B[0m  \n",
      "\u001B[?25h\u001B[34m|            fastqc\u001B[0m | Found 3 reports\n",
      "\u001B[34m|           multiqc\u001B[0m | Compressing plot data\n",
      "\u001B[34m|           multiqc\u001B[0m | Report      : multiqc_report.html\n",
      "\u001B[34m|           multiqc\u001B[0m | Data        : multiqc_data\n",
      "\u001B[34m|           multiqc\u001B[0m | MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "!multiqc ."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "-TO3DQaUoGhB",
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "Далее необходимо проанализировать выдачу fastqc, произвести фильтрацию или подрезание чтений в случае необходимости. Для примера указано использование trimmomatic, также можно использовать seqtk."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "hwoZN7AjgD1q",
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "Высокий уровень дедуплицирования ридов в ENCFF222VRH. Проведем подрезание"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "Qellza2PI7fT",
    "outputId": "58786aff-33a9-4f50-af94-fea665420e64",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "TrimmomaticSE: Started with arguments:\n",
      " -phred33 ENCFF222VRH.fastq ENCFF222VRH_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
      "Automatically using 2 threads\n",
      "java.io.FileNotFoundException: /content/TruSeq3-SE (No such file or directory)\n",
      "\tat java.io.FileInputStream.open0(Native Method)\n",
      "\tat java.io.FileInputStream.open(FileInputStream.java:195)\n",
      "\tat java.io.FileInputStream.<init>(FileInputStream.java:138)\n",
      "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
      "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
      "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
      "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
      "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
      "\tat org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)\n",
      "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:85)\n",
      "Input Reads: 58335949 Surviving: 54128905 (92.79%) Dropped: 4207044 (7.21%)\n",
      "TrimmomaticSE: Completed successfully\n"
     ]
    }
   ],
   "source": [
    "!trimmomatic SE -phred33 ENCFF222VRH.fastq ENCFF222VRH_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \\\n",
    "  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "PWBBF50gI7mI",
    "outputId": "2ebf758f-4715-4b65-b4b9-092e6c7d4ffd",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "Started analysis of ENCFF222VRH_trimmed.fastq\n",
      "Approx 5% complete for ENCFF222VRH_trimmed.fastq\n",
      "Approx 10% complete for ENCFF222VRH_trimmed.fastq\n",
      "Approx 15% complete for ENCFF222VRH_trimmed.fastq\n",
      "Approx 20% complete for ENCFF222VRH_trimmed.fastq\n",
      "Approx 25% complete for ENCFF222VRH_trimmed.fastq\n",
      "Approx 30% complete for ENCFF222VRH_trimmed.fastq\n",
      "Approx 35% complete for ENCFF222VRH_trimmed.fastq\n",
      "Approx 40% complete for ENCFF222VRH_trimmed.fastq\n",
      "Approx 45% complete for ENCFF222VRH_trimmed.fastq\n",
      "Approx 50% complete for ENCFF222VRH_trimmed.fastq\n",
      "Approx 55% complete for ENCFF222VRH_trimmed.fastq\n",
      "Approx 60% complete for ENCFF222VRH_trimmed.fastq\n",
      "Approx 65% complete for ENCFF222VRH_trimmed.fastq\n",
      "Approx 70% complete for ENCFF222VRH_trimmed.fastq\n",
      "Approx 75% complete for ENCFF222VRH_trimmed.fastq\n",
      "Approx 80% complete for ENCFF222VRH_trimmed.fastq\n",
      "Approx 85% complete for ENCFF222VRH_trimmed.fastq\n",
      "Approx 90% complete for ENCFF222VRH_trimmed.fastq\n",
      "Approx 95% complete for ENCFF222VRH_trimmed.fastq\n",
      "Analysis complete for ENCFF222VRH_trimmed.fastq\n"
     ]
    }
   ],
   "source": [
    "!./FastQC/fastqc ENCFF222VRH_trimmed.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "W9QHWBLlpMjQ",
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "## Выравнивание на хромосому\n",
    "\n",
    "Следует выбрать одну хромосому, потому что ресурсы google colab ограничены. В случае проблем, таких как Segmentation fault, error 139, следует выбрать хромосому меньшего размера.\n",
    "\n",
    "Последовательность нуклеотидов хромосомы можно скачать по адресу https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/"
   ]
  },
  {
   "cell_type": "markdown",
   "source": [
    "Взял 13 хромосому, так как не хватало места в colab"
   ],
   "metadata": {
    "id": "lZS_FlTGBXkN",
    "pycharm": {
     "name": "#%% md\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "N4JIGZOsGiY2",
    "outputId": "ce6161f9-532d-4705-bcc9-d2e021b0cd3c",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-03-07 21:13:32--  https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr13.fa.gz\n",
      "Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\n",
      "Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 31870393 (30M) [application/x-gzip]\n",
      "Saving to: ‘chr13.fa.gz’\n",
      "\n",
      "chr13.fa.gz         100%[===================>]  30.39M  11.0MB/s    in 2.8s    \n",
      "\n",
      "2022-03-07 21:13:36 (11.0 MB/s) - ‘chr13.fa.gz’ saved [31870393/31870393]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr13.fa.gz\n",
    "!gzip -d chr13.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "background_save": true,
     "base_uri": "https://localhost:8080/"
    },
    "id": "bhraQ6NdGiPN",
    "outputId": "b1c2547b-c2e7-4b17-ef85-9beeddd55c38",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Settings:\n",
      "  Output files: \"chromosome_index.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  chr13.fa\n",
      "Building a SMALL index\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:02\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:00\n",
      "bmax according to bmaxDivN setting: 24495781\n",
      "Using parameters --bmax 18371836 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 18371836 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:02\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:01\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 9.79831e+07 (target: 18371835)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 97983125 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:29\n",
      "Returning block of 97983126 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 30047611\n",
      "fchr[G]: 48886803\n",
      "fchr[T]: 67820408\n",
      "fchr[$]: 97983125\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 36855779 bytes to primary EBWT file: chromosome_index.1.bt2\n",
      "Wrote 24495788 bytes to secondary EBWT file: chromosome_index.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 97983125\n",
      "    bwtLen: 97983126\n",
      "    sz: 24495782\n",
      "    bwtSz: 24495782\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 6123946\n",
      "    offsSz: 24495784\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 510329\n",
      "    numLines: 510329\n",
      "    ebwtTotLen: 32661056\n",
      "    ebwtTotSz: 32661056\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:00:54\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:00\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:01\n",
      "  Time to reverse reference sequence: 00:00:00\n",
      "bmax according to bmaxDivN setting: 24495781\n",
      "Using parameters --bmax 18371836 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 18371836 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:02\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:01\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 9.79831e+07 (target: 18371835)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 97983125 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:29\n",
      "Returning block of 97983126 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 30047611\n",
      "fchr[G]: 48886803\n",
      "fchr[T]: 67820408\n",
      "fchr[$]: 97983125\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 36855779 bytes to primary EBWT file: chromosome_index.rev.1.bt2\n",
      "Wrote 24495788 bytes to secondary EBWT file: chromosome_index.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 97983125\n",
      "    bwtLen: 97983126\n",
      "    sz: 24495782\n",
      "    bwtSz: 24495782\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 6123946\n",
      "    offsSz: 24495784\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 510329\n",
      "    numLines: 510329\n",
      "    ebwtTotLen: 32661056\n",
      "    ebwtTotSz: 32661056\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:00:51\n",
      "CPU times: user 603 ms, sys: 99.7 ms, total: 703 ms\n",
      "Wall time: 1min 44s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!bowtie2-build chr13.fa chromosome_index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "E_usI8XUrQnR",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "!mkdir bowtie2_res"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "background_save": true
    },
    "id": "3t80UATHqFe7",
    "outputId": "e788befa-da0e-4691-b191-b79e85a76c8f",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "54128905 reads; of these:\n",
      "  54128905 (100.00%) were unpaired; of these:\n",
      "    46986193 (86.80%) aligned 0 times\n",
      "    2063611 (3.81%) aligned exactly 1 time\n",
      "    5079101 (9.38%) aligned >1 times\n",
      "13.20% overall alignment rate\n",
      "CPU times: user 15.3 s, sys: 2.12 s, total: 17.5 s\n",
      "Wall time: 53min 41s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!bowtie2  -p 2 \\\n",
    "          -x chromosome_index \\\n",
    "          -U ENCFF222VRH_trimmed.fastq \\\n",
    "          -S bowtie2_res/res_VRH.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "background_save": true
    },
    "id": "so1jTdGHoEir",
    "outputId": "a3aefe0f-ea65-4b28-838b-3041b5e44dd8",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22026319 reads; of these:\n",
      "  22026319 (100.00%) were unpaired; of these:\n",
      "    19272245 (87.50%) aligned 0 times\n",
      "    783224 (3.56%) aligned exactly 1 time\n",
      "    1970850 (8.95%) aligned >1 times\n",
      "12.50% overall alignment rate\n",
      "CPU times: user 5.96 s, sys: 782 ms, total: 6.75 s\n",
      "Wall time: 20min 41s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!bowtie2  -p 2 \\\n",
    "          -x chromosome_index \\\n",
    "          -U ENCFF134NCG.fastq \\\n",
    "          -S bowtie2_res/res_NCG.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "bVBJtUXuiJFK",
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "outputId": "32399b15-3470-4def-b736-09f1024fa18b",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "20301162 reads; of these:\n",
      "  20301162 (100.00%) were unpaired; of these:\n",
      "    17763189 (87.50%) aligned 0 times\n",
      "    724390 (3.57%) aligned exactly 1 time\n",
      "    1813583 (8.93%) aligned >1 times\n",
      "12.50% overall alignment rate\n",
      "CPU times: user 5.52 s, sys: 725 ms, total: 6.25 s\n",
      "Wall time: 19min 4s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!bowtie2  -p 2 \\\n",
    "          -x chromosome_index \\\n",
    "          -U ENCFF738ERP.fastq \\\n",
    "          -S bowtie2_res/res_ERP.sam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "ysoDrnSCrei0",
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "Проанализируйте выдачу bowtie. Почему процент выравниваний получился именно таким?\n",
    "\n",
    "Имеет смысл для дальнейшего анализа отобрать уникально картированные риды."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "TmfJKT0rmLyp",
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "Преобразование .sam в .bam (не требуется если запускать macs2 на .sam файлах)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "2PteJ-KVqFhs",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "!samtools view -h -S -b -o bowtie2_res/res_ASU.bam bowtie2_res/res_NCG.sam\n",
    "!samtools view -h -S -b -o bowtie2_res/res_ARK.bam bowtie2_res/res_ERP.sam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "eLtG7xtRup19",
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "На этом моменте можно скачать себе на компьютер все полученные .bam или .sam файлы на случай сбоя."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "XZtQrPZMmgnC",
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "## Peak calling\n",
    "\n",
    "Помимо macs2 может быть использован пакет Homer для поиска пиков, а также macs3."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "uX1ps4VDp_Ae",
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "outputId": "af10f764-a916-4a77-a0ce-fea25f6b5611",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "mkdir: cannot create directory ‘macs2’: File exists\n"
     ]
    }
   ],
   "source": [
    "!mkdir macs2\n",
    "!mkdir macs22"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "EJQOCcv2nJW_",
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "outputId": "875f0c44-8d67-4ab8-cfc0-c45c1ddbc792",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "INFO  @ Mon, 7 Mar 2022 22:50:26: \n",
      "# Command line: callpeak --broad -t bowtie2_res/res_NCG.sam -c bowtie2_res/res_NCG.sam -f SAM --outdir macs2\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['bowtie2_res/res_NCG.sam']\n",
      "# control file = ['bowtie2_res/res_NCG.sam']\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
      "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Mon, 7 Mar 2022 22:50:26: #1 read tag files... \n",
      "INFO  @ Mon, 7 Mar 2022 22:50:26: #1 read treatment tags... \n",
      "INFO  @ Mon, 7 Mar 2022 22:50:32:  5000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:50:36:  8000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:50:37:  9000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:50:45:  16000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:50:47:  18000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:50:52: #1.2 read input tags... \n",
      "INFO  @ Mon, 7 Mar 2022 22:50:56:  5000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:50:59:  8000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:50:59:  9000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:05:  16000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:07:  18000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:11: #1 tag size is determined as 76 bps \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:11: #1 tag size = 76.0 \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:11: #1  total tags in treatment: 2754074 \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:11: #1 user defined the maximum tags... \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:11: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:11: #1  tags after filtering in treatment: 2289662 \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:11: #1  Redundant rate of treatment: 0.17 \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:11: #1  total tags in control: 2754074 \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:11: #1 user defined the maximum tags... \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:11: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:11: #1  tags after filtering in control: 2289662 \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:11: #1  Redundant rate of control: 0.17 \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:11: #1 finished! \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:11: #2 Build Peak Model... \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:11: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:12: #2 number of paired peaks: 29002 \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:12: start model_add_line... \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:12: start X-correlation... \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:12: end of X-cor \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:12: #2 finished! \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:12: #2 predicted fragment length is 75 bps \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:12: #2 alternative fragment length(s) may be 75,476 bps \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:12: #2.2 Generate R script for model : macs2/NA_model.r \n",
      "WARNING @ Mon, 7 Mar 2022 22:51:12: #2 Since the d (75) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
      "WARNING @ Mon, 7 Mar 2022 22:51:12: #2 You may need to consider one of the other alternative d(s): 75,476 \n",
      "WARNING @ Mon, 7 Mar 2022 22:51:12: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:12: #3 Call peaks... \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:12: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:12: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:21: #3 Call peaks for each chromosome... \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:27: #4 Write output xls file... macs2/NA_peaks.xls \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:27: #4 Write broad peak in broadPeak format file... macs2/NA_peaks.broadPeak \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:27: #4 Write broad peak in bed12/gappedPeak format file... macs2/NA_peaks.gappedPeak \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:27: Done! \n"
     ]
    }
   ],
   "source": [
    "!macs2 callpeak --broad -t bowtie2_res/res_NCG.sam \\\n",
    "    -c bowtie2_res/res_NCG.sam \\\n",
    " \t  -f SAM \\\n",
    "\t  --outdir macs2"
   ]
  },
  {
   "cell_type": "code",
   "source": [
    "!macs2 callpeak --broad -t bowtie2_res/res_ERP.sam \\\n",
    "    -c bowtie2_res/res_ERP.sam \\\n",
    " \t  -f SAM \\\n",
    "\t  --outdir macs22"
   ],
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "n_mpnIBOBGEe",
    "outputId": "3c1974b3-0e71-4858-dc5d-76b6f44a3cf3",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "execution_count": null,
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "INFO  @ Mon, 7 Mar 2022 22:51:36: \n",
      "# Command line: callpeak --broad -t bowtie2_res/res_ERP.sam -c bowtie2_res/res_ERP.sam -f SAM --outdir macs22\n",
      "# ARGUMENTS LIST:\n",
      "# name = NA\n",
      "# format = SAM\n",
      "# ChIP-seq file = ['bowtie2_res/res_ERP.sam']\n",
      "# control file = ['bowtie2_res/res_ERP.sam']\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
      "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:36: #1 read tag files... \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:36: #1 read treatment tags... \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:41:  5000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:46:  9000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:51:  13000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:54:  16000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:51:56:  17000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:00: #1.2 read input tags... \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:06:  5000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:10:  9000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:14:  13000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:18:  16000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:19:  17000000 \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:23: #1 tag size is determined as 76 bps \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:23: #1 tag size = 76.0 \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:23: #1  total tags in treatment: 2537973 \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:23: #1 user defined the maximum tags... \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:23: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:23: #1  tags after filtering in treatment: 2132856 \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:23: #1  Redundant rate of treatment: 0.16 \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:23: #1  total tags in control: 2537973 \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:23: #1 user defined the maximum tags... \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:23: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:23: #1  tags after filtering in control: 2132856 \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:23: #1  Redundant rate of control: 0.16 \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:23: #1 finished! \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:23: #2 Build Peak Model... \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:23: #2 looking for paired plus/minus strand peaks... \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:24: #2 number of paired peaks: 27228 \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:24: start model_add_line... \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:24: start X-correlation... \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:24: end of X-cor \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:24: #2 finished! \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:24: #2 predicted fragment length is 75 bps \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:24: #2 alternative fragment length(s) may be 75 bps \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:24: #2.2 Generate R script for model : macs22/NA_model.r \n",
      "WARNING @ Mon, 7 Mar 2022 22:52:24: #2 Since the d (75) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
      "WARNING @ Mon, 7 Mar 2022 22:52:24: #2 You may need to consider one of the other alternative d(s): 75 \n",
      "WARNING @ Mon, 7 Mar 2022 22:52:24: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:24: #3 Call peaks... \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:24: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:24: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:33: #3 Call peaks for each chromosome... \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:38: #4 Write output xls file... macs22/NA_peaks.xls \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:38: #4 Write broad peak in broadPeak format file... macs22/NA_peaks.broadPeak \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:38: #4 Write broad peak in bed12/gappedPeak format file... macs22/NA_peaks.gappedPeak \n",
      "INFO  @ Mon, 7 Mar 2022 22:52:38: Done! \n"
     ]
    }
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "HrNwxs7ArPz7",
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "outputId": "54e8147b-d835-408b-b282-f932688a7b18",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "--2022-03-07 22:57:37--  https://www.encodeproject.org/files/ENCFF151QZZ/@@download/ENCFF151QZZ.bed.gz\n",
      "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
      "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
      "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
      "Location: https://encode-public.s3.amazonaws.com/2017/02/01/cc32e3a2-58ff-47fd-b40f-3374a79274b4/ENCFF151QZZ.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF151QZZ.bed.gz&AWSAccessKeyId=ASIATGZNGCNX3NA2O5HM&Signature=dko7HgQzfef4NPF3KPaCg%2FLuVeU%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEE8aCXVzLXdlc3QtMiJHMEUCICyaQEmda7%2FZDW5kv2LViJ0%2B1ShyD5VkragvwW3U2dTwAiEAmy6v5bo90miTlt3%2FpclUJNeGgGaxiMxNF3C52iYhTagqgwQI%2BP%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDDlnhYCtCDKfyE0vkyrXA%2FI0uBTsje1EIxzWLPgt5IZx5qIYWj%2BbYtGGw0Z1k2seXlhDzs%2BCNXv%2BJ%2Bma5L0DAw6vy40guE4GujepUCu1rZm7gHFAGx8i9gvaIaiLPfJXaEYXXHfp4mO%2BX8IFdKoCBA9lUooCNKFseDPzSWqUKO4L12p5OjBTGV%2FrgbQgi7w6CMY3%2Bc5Vijke6bh5nd8QqtI0w3%2FlAvDitcOIR2bb1Nt7Otw%2BIe8mUWHgKnkwQTV6dbJnSodSl9469pGW4eq8DGHuD5WZHXGzAZ7BNAdoweQPbSmMlu7QmQpYWDQ4HGldJcUsxcE%2FwPbZFH5dUK7g4sulQmSNAkVkp9g4L0j9w4Y3Z53gH5INfN4CavTeypvMC9OrG8ARpYvK39gaSglDPdEXMo0FQvTuufvEdJFEC4jImwuqF0NLGuRwl7J%2BVnmEPeHmjzI5RdUV2SZJH1YQYS4x2k4OMpK2FNu9gFoQ44E4hkI1WyD03BpZLy2LcrhtzshW9%2FFSd4Fc5b5YzaO44tMKtnNj5009D%2BK%2Fuad8uAJdz9syuedtuzRNRvM8KTBsj3DQtyqGktJPi9e%2Ftxi77O0KPBjMVKZsoZ8IHIAPNSrj81JocRcs1htumGr7Fs0H0cNMSxtrfTCE9PySBjqlAYKgnUzxRXIjdJTi8sQxJURE2yAeoLC%2B96zYlhLkvq0bgQa1b6C9W1HEJoe%2FDpMY9ZsT8EfikQDcZPxKdNzduME61D5Pkic9iYfDY4OM706kjh8xBPGyqY6yG%2BPU12hixBKoTb91MAmsuXWERhvJClOHj%2FtUPYkiArJJWXB0AUWk6cnh1zvfR0xRHI2wZNIBaYtJD%2FvYgIbaN5NNwVbutv%2F%2B%2BVqlAA%3D%3D&Expires=1650538657 [following]\n",
      "--2022-03-07 22:57:38--  https://encode-public.s3.amazonaws.com/2017/02/01/cc32e3a2-58ff-47fd-b40f-3374a79274b4/ENCFF151QZZ.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF151QZZ.bed.gz&AWSAccessKeyId=ASIATGZNGCNX3NA2O5HM&Signature=dko7HgQzfef4NPF3KPaCg%2FLuVeU%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEE8aCXVzLXdlc3QtMiJHMEUCICyaQEmda7%2FZDW5kv2LViJ0%2B1ShyD5VkragvwW3U2dTwAiEAmy6v5bo90miTlt3%2FpclUJNeGgGaxiMxNF3C52iYhTagqgwQI%2BP%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDDlnhYCtCDKfyE0vkyrXA%2FI0uBTsje1EIxzWLPgt5IZx5qIYWj%2BbYtGGw0Z1k2seXlhDzs%2BCNXv%2BJ%2Bma5L0DAw6vy40guE4GujepUCu1rZm7gHFAGx8i9gvaIaiLPfJXaEYXXHfp4mO%2BX8IFdKoCBA9lUooCNKFseDPzSWqUKO4L12p5OjBTGV%2FrgbQgi7w6CMY3%2Bc5Vijke6bh5nd8QqtI0w3%2FlAvDitcOIR2bb1Nt7Otw%2BIe8mUWHgKnkwQTV6dbJnSodSl9469pGW4eq8DGHuD5WZHXGzAZ7BNAdoweQPbSmMlu7QmQpYWDQ4HGldJcUsxcE%2FwPbZFH5dUK7g4sulQmSNAkVkp9g4L0j9w4Y3Z53gH5INfN4CavTeypvMC9OrG8ARpYvK39gaSglDPdEXMo0FQvTuufvEdJFEC4jImwuqF0NLGuRwl7J%2BVnmEPeHmjzI5RdUV2SZJH1YQYS4x2k4OMpK2FNu9gFoQ44E4hkI1WyD03BpZLy2LcrhtzshW9%2FFSd4Fc5b5YzaO44tMKtnNj5009D%2BK%2Fuad8uAJdz9syuedtuzRNRvM8KTBsj3DQtyqGktJPi9e%2Ftxi77O0KPBjMVKZsoZ8IHIAPNSrj81JocRcs1htumGr7Fs0H0cNMSxtrfTCE9PySBjqlAYKgnUzxRXIjdJTi8sQxJURE2yAeoLC%2B96zYlhLkvq0bgQa1b6C9W1HEJoe%2FDpMY9ZsT8EfikQDcZPxKdNzduME61D5Pkic9iYfDY4OM706kjh8xBPGyqY6yG%2BPU12hixBKoTb91MAmsuXWERhvJClOHj%2FtUPYkiArJJWXB0AUWk6cnh1zvfR0xRHI2wZNIBaYtJD%2FvYgIbaN5NNwVbutv%2F%2B%2BVqlAA%3D%3D&Expires=1650538657\n",
      "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.160.30\n",
      "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.160.30|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 1521843 (1.5M) [binary/octet-stream]\n",
      "Saving to: ‘ENCFF151QZZ.bed.gz’\n",
      "\n",
      "ENCFF151QZZ.bed.gz  100%[===================>]   1.45M  1.50MB/s    in 1.0s    \n",
      "\n",
      "2022-03-07 22:57:39 (1.50 MB/s) - ‘ENCFF151QZZ.bed.gz’ saved [1521843/1521843]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://www.encodeproject.org/files/ENCFF151QZZ/@@download/ENCFF151QZZ.bed.gz\n",
    "!gzip -d ENCFF151QZZ.bed.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "MYEGEbQjvEwY",
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "## Сравнение результатов\n",
    "\n",
    "Сравниваем те пики, которые мы получили, с пиками, которые приведены в ENCODE (важно, чтобы версии генома hg38 или hg19) для .bed файла из ENCODE и той хромосомы, которую скачивали выше, сопадали.\n",
    "\n",
    "Проанализируйте полученные результаты и приведите свои рассуждения в README.md. Как можно объяснить различия в количестве пересечений?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "_1EAygSCvc8a",
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "outputId": "1548d65f-4729-4a0e-cb99-745d90ae0ebb",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "\n",
      "Generating a 2-way \"venn\" diagram. Please wait...\n",
      "\n",
      "\n",
      "Done! Please check your results @ /content/Intervene_results. \n",
      "Thank you for using Intervene!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!intervene venn -i macs2/NA_peaks.broadPeak ENCFF151QZZ.bed --filenames"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "7qqn3n2cRdkt",
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "outputId": "f12bfd4e-5b5e-487f-f210-5c657e0f3e37",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "\n",
      "Generating a 2-way \"venn\" diagram. Please wait...\n",
      "\n",
      "\n",
      "Done! Please check your results @ /content/Intervene_results. \n",
      "Thank you for using Intervene!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!intervene venn -i ENCFF151QZZ.bed macs2/NA_peaks.broadPeak --filenames"
   ]
  },
  {
   "cell_type": "code",
   "source": [
    "!intervene venn -i macs22/NA_peaks.broadPeak ENCFF151QZZ.bed --filenames"
   ],
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "bNxTmRPXDdol",
    "outputId": "3bf7d3df-6d78-4436-ea81-2f9eddee900f",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "execution_count": null,
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "\n",
      "Generating a 2-way \"venn\" diagram. Please wait...\n",
      "\n",
      "\n",
      "Done! Please check your results @ /content/Intervene_results. \n",
      "Thank you for using Intervene!\n",
      "\n"
     ]
    }
   ]
  },
  {
   "cell_type": "code",
   "source": [
    "!intervene venn -i ENCFF151QZZ.bed macs22/NA_peaks.broadPeak --filenames"
   ],
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "8ZP9xWhLDtNr",
    "outputId": "9a6116fe-8f4b-4fa8-d04b-aa9bff34a4c1",
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "execution_count": null,
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "\n",
      "Generating a 2-way \"venn\" diagram. Please wait...\n",
      "\n",
      "\n",
      "Done! Please check your results @ /content/Intervene_results. \n",
      "Thank you for using Intervene!\n",
      "\n"
     ]
    }
   ]
  }
 ],
 "metadata": {
  "colab": {
   "collapsed_sections": [],
   "name": "Копия блокнота \"ChIP-seq on histones.ipynb\"",
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3",
   "name": "python3"
  },
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}